// @bgroup/ui/beyond-drag-and-drop/beyond-drag-and-drop.sjs.js
System.register(["@beyond-js/widgets/render", "@beyond-js/kernel/bundle", "@beyond-js/react-widgets/page", "react", "@bgroup/ui/header", "@bgroup/ui/copyable-implementation", "@bgroup/ui/implementation-result", "@bgroup/ui/drag-and-drop"], function (_export, _context) {
  "use strict";

  var dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5, dependency_6, dependency_7, __Bundle, __pkg, ims, Controller, View, __beyond_pkg, hmr;
  _export({
    Controller: void 0,
    View: void 0
  });
  return {
    setters: [function (_beyondJsWidgetsRender) {
      dependency_0 = _beyondJsWidgetsRender;
    }, function (_beyondJsKernelBundle) {
      dependency_1 = _beyondJsKernelBundle;
    }, function (_beyondJsReactWidgetsPage) {
      dependency_2 = _beyondJsReactWidgetsPage;
    }, function (_react) {
      dependency_3 = _react;
    }, function (_bgroupUiHeader) {
      dependency_4 = _bgroupUiHeader;
    }, function (_bgroupUiCopyableImplementation) {
      dependency_5 = _bgroupUiCopyableImplementation;
    }, function (_bgroupUiImplementationResult) {
      dependency_6 = _bgroupUiImplementationResult;
    }, function (_bgroupUiDragAndDrop) {
      dependency_7 = _bgroupUiDragAndDrop;
    }],
    execute: function () {
      ({
        Bundle: __Bundle
      } = dependency_1);
      __pkg = new __Bundle({
        "module": {
          "vspecifier": "@bgroup/ui@0.0.34/beyond-drag-and-drop"
        },
        "type": "widget"
      }, _context.meta.url).package();
      ;
      __pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0], ['@beyond-js/react-widgets/page', dependency_2], ['react', dependency_3], ['@bgroup/ui/header', dependency_4], ['@bgroup/ui/copyable-implementation', dependency_5], ['@bgroup/ui/implementation-result', dependency_6], ['@bgroup/ui/drag-and-drop', dependency_7]]);
      brequire('@beyond-js/widgets/render').widgets.register([{
        "name": "beyond-drag-and-drop",
        "vspecifier": "@bgroup/ui@0.0.34/beyond-drag-and-drop",
        "is": "page",
        "route": "/components/beyond-drag-and-drop"
      }]);
      ims = new Map();
      /****************************
      INTERNAL MODULE: ./controller
      ****************************/
      ims.set('./controller', {
        hash: 3638196390,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Controller = void 0;
          var _page = require("@beyond-js/react-widgets/page");
          var _views = require("./views");
          /*bundle*/
          class Controller extends _page.PageReactWidgetController {
            get Widget() {
              return _views.View;
            }
          }
          exports.Controller = Controller;
        }
      });

      /**************************************
      INTERNAL MODULE: ./views/implementation
      **************************************/

      ims.set('./views/implementation', {
        hash: 2577044970,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.implementationPhoto = exports.implementation = void 0;
          const implementation = `
export function DragAndDropImplementation() {
	function onUpload(files: File[]): void {
		//your logic
	};
	const formats = ['pdf']
	return (
		<div className="results-form">
			<DragAndDrop
				onUpload={onUpload}
				count={2}
				formats={formats}
			/>
		</div>
	);
}`;
          exports.implementation = implementation;
          const implementationPhoto = `
export function DragAndDropImplementationPhoto() {
	function onUpload(files: File[]): void {
		//your logic
	};

	return (
		<div className="results-form">
			<DragAndDrop
				onUpload={onUpload}
				userPhoto
			/>
		</div>
	);
}`;
          exports.implementationPhoto = implementationPhoto;
        }
      });

      /*****************************
      INTERNAL MODULE: ./views/index
      *****************************/

      ims.set('./views/index', {
        hash: 3853311252,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.View = View;
          var React = require("react");
          var _header = require("@bgroup/ui/header");
          var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
          var _results = require("./results");
          var _implementation = require("./implementation");
          var _implementationResult = require("@bgroup/ui/implementation-result");
          var _resultsUserPhoto = require("./results-user-photo");
          /*bundle*/
          function View() {
            return React.createElement("div", {
              className: "view"
            }, React.createElement(_header.Header, {
              componentName: "DragAndDrop"
            }), React.createElement("div", {
              className: "component-info"
            }, React.createElement("h3", null, "Import: "), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { DragAndDrop } from '@bgroup/ui/drag-and-drop';`)), React.createElement("div", {
              className: "component-implementation"
            }, React.createElement("div", {
              className: "component-result"
            }, React.createElement("div", null, React.createElement("h3", null, "Case upload file:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementation
            }, React.createElement(_results.Results, null))), React.createElement("div", {
              className: "component-detail"
            }, React.createElement("h4", null, "Component information:"), React.createElement("div", null, React.createElement("p", null, "- Receives ", React.createElement("code", null, "count"), " to define the maximum amount of files that can be loaded if you want to define no limit pass to count", React.createElement("code", null, "Infinity"), " ", React.createElement("br", null), "- Receives ", React.createElement("code", null, "formats"), " which is an array of strings with the available formats that can be loaded. ", React.createElement("br", null), "- Receives ", React.createElement("code", null, "onUpload"), " which is a function that receives the uploaded files.")))), React.createElement("div", {
              className: "component-result"
            }, React.createElement("div", {
              className: "w-100"
            }, React.createElement("h3", null, "Case upload file:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementationPhoto
            }, React.createElement(_resultsUserPhoto.ResultsUserPhoto, null))), React.createElement("div", {
              className: "component-detail"
            }, React.createElement("h4", null, "Component information:"), React.createElement("div", null, React.createElement("p", null, "- Receives ", React.createElement("code", null, "textUserPhoto"), " in case you want to change the text on the component ", React.createElement("br", null), "- Receives ", React.createElement("code", null, "onUpload"), " which is a function that receives the uploaded files. ", React.createElement("br", null), "- By default only one file and image format is supported."))))));
          }
        }
      });

      /******************************************
      INTERNAL MODULE: ./views/results-user-photo
      ******************************************/

      ims.set('./views/results-user-photo', {
        hash: 2104309916,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultsUserPhoto = ResultsUserPhoto;
          var React = require("react");
          var _dragAndDrop = require("@bgroup/ui/drag-and-drop");
          function ResultsUserPhoto() {
            function onUpload(files) {
              //your logic
            }
            ;
            return React.createElement("div", {
              className: "results-form"
            }, React.createElement(_dragAndDrop.DragAndDrop, {
              onUpload: onUpload,
              userPhoto: true
            }));
          }
        }
      });

      /*******************************
      INTERNAL MODULE: ./views/results
      *******************************/

      ims.set('./views/results', {
        hash: 1091441419,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Results = Results;
          var React = require("react");
          var _dragAndDrop = require("@bgroup/ui/drag-and-drop");
          function Results() {
            function onUpload(files) {
              //your logic
            }
            ;
            const formats = ['pdf'];
            return React.createElement("div", {
              className: "results-form"
            }, React.createElement(_dragAndDrop.DragAndDrop, {
              onUpload: onUpload,
              count: 2,
              formats: formats
            }));
          }
        }
      });
      __pkg.exports.descriptor = [{
        "im": "./controller",
        "from": "Controller",
        "name": "Controller"
      }, {
        "im": "./views/index",
        "from": "View",
        "name": "View"
      }];
      // Module exports
      __pkg.exports.process = function ({
        require,
        prop,
        value
      }) {
        (require || prop === 'Controller') && _export("Controller", Controller = require ? require('./controller').Controller : value);
        (require || prop === 'View') && _export("View", View = require ? require('./views/index').View : value);
      };
      _export("__beyond_pkg", __beyond_pkg = __pkg);
      _export("hmr", hmr = new function () {
        this.on = (event, listener) => void 0;
        this.off = (event, listener) => void 0;
      }());
      __pkg.initialise(ims);
    }
  };
});
