// @bgroup/ui/link/link.sjs.js
System.register(["@beyond-js/kernel/bundle", "react", "@beyond-js/kernel/routing"], function (_export, _context) {
  "use strict";

  var dependency_0, dependency_1, dependency_2, __Bundle, __pkg, ims, Link, Elink, __beyond_pkg, hmr;
  _export({
    Link: void 0,
    Elink: void 0
  });
  return {
    setters: [function (_beyondJsKernelBundle) {
      dependency_0 = _beyondJsKernelBundle;
    }, function (_react) {
      dependency_1 = _react;
    }, function (_beyondJsKernelRouting) {
      dependency_2 = _beyondJsKernelRouting;
    }],
    execute: function () {
      ({
        Bundle: __Bundle
      } = dependency_0);
      __pkg = new __Bundle({
        "module": {
          "vspecifier": "@bgroup/ui@0.0.34/link"
        },
        "type": "code"
      }, _context.meta.url).package();
      ;
      __pkg.dependencies.update([['react', dependency_1], ['@beyond-js/kernel/routing', dependency_2]]);
      ims = new Map();
      /***********************
      INTERNAL MODULE: ./index
      ***********************/
      ims.set('./index', {
        hash: 3282325095,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Elink = Elink;
          exports.Link = Link;
          var React = require("react");
          var _routing = require("@beyond-js/kernel/routing");
          /*bundle*/
          function Link(props) {
            const onClick = event => {
              event.preventDefault();
              event.stopPropagation();
              if (props.onClick && typeof props.onClick === "function") {
                props.onClick(event);
              }
              _routing.routing.pushState(props.href);
            };
            const properties = {
              ...props
            };
            delete properties.href;
            delete properties.onClick;
            return React.createElement("a", {
              onClick: onClick,
              ...properties
            }, props.children);
          }
          /*bundle*/
          function Elink(props) {
            const {
              href
            } = props;
            delete props.href;
            return React.createElement("a", {
              href: href,
              target: "_blank",
              ...props,
              "data-algo": "algo"
            }, props.children);
          }
        }
      });
      __pkg.exports.descriptor = [{
        "im": "./index",
        "from": "Link",
        "name": "Link"
      }, {
        "im": "./index",
        "from": "Elink",
        "name": "Elink"
      }];
      // Module exports
      __pkg.exports.process = function ({
        require,
        prop,
        value
      }) {
        (require || prop === 'Link') && _export("Link", Link = require ? require('./index').Link : value);
        (require || prop === 'Elink') && _export("Elink", Elink = require ? require('./index').Elink : value);
      };
      _export("__beyond_pkg", __beyond_pkg = __pkg);
      _export("hmr", hmr = new function () {
        this.on = (event, listener) => void 0;
        this.off = (event, listener) => void 0;
      }());
      __pkg.initialise(ims);
    }
  };
});
