// @bgroup/ui/beyond-image/beyond-image.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/react-widgets/page';
import * as dependency_3 from 'react';
import * as dependency_4 from '@bgroup/ui/header';
import * as dependency_5 from '@bgroup/ui/copyable-implementation';
import * as dependency_6 from '@bgroup/ui/image';
import * as dependency_7 from '@beyond-js/kernel/styles';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/beyond-image"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/react-widgets/page', dependency_2],['react', dependency_3],['@bgroup/ui/header', dependency_4],['@bgroup/ui/copyable-implementation', dependency_5],['@bgroup/ui/image', dependency_6],['@beyond-js/kernel/styles', dependency_7]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"beyond-image","vspecifier":"@bgroup/ui@0.0.34/beyond-image","is":"page","route":"/components/beyond-image"}]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-image')


const ims = new Map();

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 3638196390, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _page = require("@beyond-js/react-widgets/page");
var _views = require("./views");
/*bundle*/class Controller extends _page.PageReactWidgetController {
  get Widget() {
    return _views.View;
  }
}
exports.Controller = Controller;
}});

/**************************************
INTERNAL MODULE: ./views/implementation
**************************************/

ims.set('./views/implementation', {hash: 497111328, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.implementation = void 0;
const implementation = `
<BeyondImage src="https://th.bing.com/th/id/R.676e186b253fa27707e3a5e288134d5a?rik=9%2bGaUnKmaIlMEA&pid=ImgRaw&r=0" alt="Brasil vencio a Argentina" />;`;
exports.implementation = implementation;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 1705018848, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.View = View;
var React = require("react");
var _header = require("@bgroup/ui/header");
var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
var _results = require("./results");
var _implementation = require("./implementation");
/*bundle*/function View() {
  return React.createElement("div", {
    className: "view beyond-image-view"
  }, React.createElement(_header.Header, {
    componentName: "Image"
  }), React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Import: "), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { BeyondImage } from '@bgroup/ui/image';`)), React.createElement("div", {
    className: 'component-implementation'
  }, React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Implementation: "), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation), React.createElement("div", {
    className: 'component-detail'
  }, React.createElement("h4", null, "Additional Information:"), React.createElement("div", null, React.createElement("p", null, "This component detects when the image is loading or when the image could not be loaded in both cases a different html class is set, for the loading case ", React.createElement("code", null, "beyond-element-image-preload"), "\u00A0 and for the error case", React.createElement("code", null, "beyond-element-image-error"))))), React.createElement("div", {
    className: "component-result"
  }, React.createElement("h3", null, "Result:"), React.createElement(_results.Results, null))));
}
}});

/*******************************
INTERNAL MODULE: ./views/results
*******************************/

ims.set('./views/results', {hash: 1409976199, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Results = Results;
var React = require("react");
var _image = require("@bgroup/ui/image");
function Results() {
  return React.createElement(_image.BeyondImage, {
    src: "https://images.freeimages.com/images/large-previews/19f/the-manor-1642006.jpg",
    alt: "Brasil vencio a Argentina"
  });
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"},{"im":"./views/index","from":"View","name":"View"}];

export let Controller, View;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);
    (require || prop === 'View') && (View = require ? require('./views/index').View : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
