// @bgroup/ui/implementation-result/implementation-result.sjs.js
System.register(["@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react", "@bgroup/ui/icons", "prism-react-renderer", "@bgroup/ui/toast", "@bgroup/ui/tabs"], function (_export, _context) {
  "use strict";

  var dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5, dependency_6, __Bundle, __pkg, ims, ImplementationResult, __beyond_pkg, hmr;
  _export("ImplementationResult", void 0);
  return {
    setters: [function (_beyondJsKernelBundle) {
      dependency_0 = _beyondJsKernelBundle;
    }, function (_beyondJsKernelStyles) {
      dependency_1 = _beyondJsKernelStyles;
    }, function (_react2) {
      dependency_2 = _react2;
    }, function (_bgroupUiIcons) {
      dependency_3 = _bgroupUiIcons;
    }, function (_prismReactRenderer2) {
      dependency_4 = _prismReactRenderer2;
    }, function (_bgroupUiToast) {
      dependency_5 = _bgroupUiToast;
    }, function (_bgroupUiTabs) {
      dependency_6 = _bgroupUiTabs;
    }],
    execute: function () {
      ({
        Bundle: __Bundle
      } = dependency_0);
      __pkg = new __Bundle({
        "module": {
          "vspecifier": "@bgroup/ui@0.0.34/implementation-result"
        },
        "type": "code"
      }, _context.meta.url).package();
      ;
      __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2], ['@bgroup/ui/icons', dependency_3], ['prism-react-renderer', dependency_4], ['@bgroup/ui/toast', dependency_5], ['@bgroup/ui/tabs', dependency_6]]);
      brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/implementation-result');
      ims = new Map();
      /**********************
      INTERNAL MODULE: ./code
      **********************/
      ims.set('./code', {
        hash: 2939750237,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.CopyableImplementation = CopyableImplementation;
          var React = require("react");
          var _icons = require("@bgroup/ui/icons");
          var _prismReactRenderer = require("prism-react-renderer");
          var _toast = require("@bgroup/ui/toast");
          function CopyableImplementation({
            code
          }) {
            function copy() {
              navigator.clipboard.writeText(code);
              _toast.toast.success("Copied implementation");
            }
            return React.createElement("div", {
              className: "copyable-content"
            }, React.createElement("div", {
              className: "code"
            }, React.createElement(_prismReactRenderer.default, {
              ..._prismReactRenderer.defaultProps,
              code: code,
              language: "jsx"
            }, ({
              className,
              style,
              tokens,
              getLineProps,
              getTokenProps
            }) => React.createElement("code", {
              className: className,
              style: style
            }, tokens.map((line, i) => React.createElement("div", {
              ...getLineProps({
                line,
                key: i
              })
            }, line.map((token, key) => React.createElement("span", {
              ...getTokenProps({
                token,
                key
              })
            }))))))), React.createElement(_icons.BeyondIconButton, {
              icon: "copy",
              onClick: copy,
              className: "copy-button"
            }));
          }
        }
      });

      /**********************
      INTERNAL MODULE: ./view
      **********************/

      ims.set('./view', {
        hash: 2724580302,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ImplementationResult = ImplementationResult;
          var _react = require("react");
          var _tabs = require("@bgroup/ui/tabs");
          var _icons = require("@bgroup/ui/icons");
          var _code = require("./code");
          /*bundle*/
          function ImplementationResult({
            code,
            children
          }) {
            const [tab, setTab] = _react.default.useState(0);
            return _react.default.createElement(_tabs.BeyondTabs, {
              selected: tab
            }, _react.default.createElement(_tabs.Tabs, {
              selected: tab
            }, _react.default.createElement(_icons.BeyondIconButton, {
              "data-tooltip": "Result",
              className: "tooltip",
              icon: "result",
              onClick: () => setTab(0)
            }), _react.default.createElement(_icons.BeyondIconButton, {
              "data-tooltip": "Code",
              className: "tooltip",
              icon: "code",
              onClick: () => setTab(1)
            })), _react.default.createElement(_tabs.TabsContent, null, children, _react.default.createElement(_code.CopyableImplementation, {
              code: code
            })));
          }
        }
      });
      __pkg.exports.descriptor = [{
        "im": "./view",
        "from": "ImplementationResult",
        "name": "ImplementationResult"
      }];
      // Module exports
      __pkg.exports.process = function ({
        require,
        prop,
        value
      }) {
        (require || prop === 'ImplementationResult') && _export("ImplementationResult", ImplementationResult = require ? require('./view').ImplementationResult : value);
      };
      _export("__beyond_pkg", __beyond_pkg = __pkg);
      _export("hmr", hmr = new function () {
        this.on = (event, listener) => void 0;
        this.off = (event, listener) => void 0;
      }());
      __pkg.initialise(ims);
    }
  };
});
