// @bgroup/ui/beyond-code-verification/beyond-code-verification.amd.js
define(["exports", "module", "@beyond-js/widgets/render", "@beyond-js/kernel/bundle", "@beyond-js/react-widgets/page", "@bgroup/ui/reactive-model", "react", "@bgroup/ui/header", "@bgroup/ui/copyable-implementation", "@bgroup/ui/hooks", "@bgroup/ui/code-verification", "@beyond-js/kernel/styles"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5, dependency_6, dependency_7, dependency_8, dependency_9) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.View = _exports.Controller = void 0;
  const {
    Bundle: __Bundle
  } = dependency_1;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "@bgroup/ui@0.0.34/beyond-code-verification"
    },
    "type": "widget"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0], ['@beyond-js/react-widgets/page', dependency_2], ['@bgroup/ui/reactive-model', dependency_3], ['react', dependency_4], ['@bgroup/ui/header', dependency_5], ['@bgroup/ui/copyable-implementation', dependency_6], ['@bgroup/ui/hooks', dependency_7], ['@bgroup/ui/code-verification', dependency_8], ['@beyond-js/kernel/styles', dependency_9]]);
  brequire('@beyond-js/widgets/render').widgets.register([{
    "name": "beyond-code-verification",
    "vspecifier": "@bgroup/ui@0.0.34/beyond-code-verification",
    "is": "page",
    "route": "/components/beyond-code-verification"
  }]);
  brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-code-verification');
  const ims = new Map();

  /****************************
  INTERNAL MODULE: ./controller
  ****************************/

  ims.set('./controller', {
    hash: 3999273080,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Controller = void 0;
      var _page = require("@beyond-js/react-widgets/page");
      var _views = require("./views");
      var _model = require("./model");
      /*bundle*/
      class Controller extends _page.PageReactWidgetController {
        get Widget() {
          return _views.View;
        }
        #model;
        createStore() {
          this.#model = new _model.Model();
          return this.#model;
        }
        hide() {
          this.#model.hide();
        }
      }
      exports.Controller = Controller;
    }
  });

  /***********************
  INTERNAL MODULE: ./model
  ***********************/

  ims.set('./model', {
    hash: 1336825640,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Model = void 0;
      var _reactiveModel = require("@bgroup/ui/reactive-model");
      class Model extends _reactiveModel.ReactiveModel {
        #count = 0;
        get count() {
          return this.#count;
        }
        hide = () => {
          this.#count = this.#count + 1;
          this.triggerEvent("hide");
        };
      }
      exports.Model = Model;
    }
  });

  /**************************************
  INTERNAL MODULE: ./views/implementation
  **************************************/

  ims.set('./views/implementation', {
    hash: 3646672315,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.implementation = void 0;
      const implementation = `
export function CodeVerificationImplementation() {

	const [code, setCode] = React.useState<string>("");

	const onCodeFull = (code: string): void => setCode(code);
	
	return (
	  <InputCode
		className="code-result"
		length={6}
		onCodeFull={onCodeFull}
		onlyNumber={false}
	  />
	);
}
`;
      exports.implementation = implementation;
    }
  });

  /*****************************
  INTERNAL MODULE: ./views/index
  *****************************/

  ims.set('./views/index', {
    hash: 4212401518,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.View = View;
      var React = require("react");
      var _header = require("@bgroup/ui/header");
      var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
      var _results = require("./results");
      var _implementation = require("./implementation");
      var _hooks = require("@bgroup/ui/hooks");
      /*bundle*/
      function View({
        store
      }) {
        const [reset, setReset] = React.useState(0);
        (0, _hooks.useBinder)([store], () => {
          setReset(store.count);
        }, "hide");
        return React.createElement("div", {
          className: "view beyond-image-view"
        }, React.createElement(_header.Header, {
          componentName: "Verification Code"
        }), React.createElement("div", {
          className: "component-info"
        }, React.createElement("h3", null, "Import: "), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { InputCode } from "@bgroup/ui/code-verification";`)), React.createElement("div", {
          className: 'component-implementation'
        }, React.createElement("div", {
          className: "component-info"
        }, React.createElement("h3", null, "Implementation: "), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation)), React.createElement("div", {
          className: "component-result"
        }, React.createElement("h3", null, "Result:"), React.createElement(_results.Result, {
          reset: reset
        }))));
      }
    }
  });

  /*******************************
  INTERNAL MODULE: ./views/results
  *******************************/

  ims.set('./views/results', {
    hash: 2301392751,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Result = Result;
      var React = require("react");
      var _codeVerification = require("@bgroup/ui/code-verification");
      function Result({
        reset
      }) {
        const [code, setCode] = React.useState("");
        const onCodeFull = code => setCode(code);
        return React.createElement(_codeVerification.InputCode, {
          className: "code-result",
          length: 6,
          onCodeFull: onCodeFull,
          onlyNumber: false,
          reset: reset
        });
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./controller",
    "from": "Controller",
    "name": "Controller"
  }, {
    "im": "./views/index",
    "from": "View",
    "name": "View"
  }];
  let Controller, View;

  // Module exports
  _exports.View = View;
  _exports.Controller = Controller;
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'Controller') && (_exports.Controller = Controller = require ? require('./controller').Controller : value);
    (require || prop === 'View') && (_exports.View = View = require ? require('./views/index').View : value);
  };
  const __beyond_pkg = __pkg;
  _exports.__beyond_pkg = __beyond_pkg;
  const hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  _exports.hmr = hmr;
  __pkg.initialise(ims);
});
