// @bgroup/ui/beyond-scroll-container/beyond-scroll-container.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/react-widgets/page';
import * as dependency_3 from 'react';
import * as dependency_4 from '@bgroup/ui/header';
import * as dependency_5 from '@bgroup/ui/copyable-implementation';
import * as dependency_6 from '@bgroup/ui/perfect-scrollbar';
import * as dependency_7 from '@beyond-js/kernel/styles';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/beyond-scroll-container"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/react-widgets/page', dependency_2],['react', dependency_3],['@bgroup/ui/header', dependency_4],['@bgroup/ui/copyable-implementation', dependency_5],['@bgroup/ui/perfect-scrollbar', dependency_6],['@beyond-js/kernel/styles', dependency_7]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"beyond-scroll-container","vspecifier":"@bgroup/ui@0.0.34/beyond-scroll-container","is":"page","route":"/components/beyond-scroll-container"}]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-scroll-container')


const ims = new Map();

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 3638196390, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _page = require("@beyond-js/react-widgets/page");
var _views = require("./views");
/*bundle*/class Controller extends _page.PageReactWidgetController {
  get Widget() {
    return _views.View;
  }
}
exports.Controller = Controller;
}});

/**************************************
INTERNAL MODULE: ./views/implementation
**************************************/

ims.set('./views/implementation', {hash: 2062646560, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.implementation = void 0;
const implementation = `
export function ImplementationPerfectScroll() {
  const text = 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptates, quis veniam.';
	const output = [];
	for (var i = 0; i < 300; i++) {
		output.push(text)
	  }
	return (
		<BeyondScrollContainer
			className="aside"
			data-perfect-scrollbar=""
			data-suppress-scroll-x="true"
		>
			<div>
				{output.join('')}
			</div>
			<div className="ps__rail-x">
				<div className="ps__thumb-x" tabIndex={0} />
			</div>
			<div className="ps__rail-y">
				<div className=" ps__thumb-y" tabIndex={0} />
			</div>
		</BeyondScrollContainer>
  }
`;
exports.implementation = implementation;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 2571637783, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.View = View;
var React = require("react");
var _header = require("@bgroup/ui/header");
var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
var _results = require("./results");
var _implementation = require("./implementation");
/*bundle*/function View() {
  return React.createElement("div", {
    className: "view beyond-image-view"
  }, React.createElement(_header.Header, {
    componentName: "Perfect Scroll"
  }), React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Import: "), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { BeyondScrollContainer } from "@bgroup/ui/perfect-scrollbar";`)), React.createElement("div", {
    className: 'component-implementation'
  }, React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Implementation: "), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation)), React.createElement("div", {
    className: "component-result"
  }, React.createElement("h3", null, "Result:"), React.createElement(_results.Results, null))));
}
}});

/*******************************
INTERNAL MODULE: ./views/results
*******************************/

ims.set('./views/results', {hash: 462025889, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Results = Results;
var React = require("react");
var _perfectScrollbar = require("@bgroup/ui/perfect-scrollbar");
function Results() {
  const text = 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptates, quis veniam.';
  const output = [];
  for (var i = 0; i < 300; i++) {
    output.push(text);
  }
  return React.createElement(_perfectScrollbar.BeyondScrollContainer, {
    className: "aside",
    "data-perfect-scrollbar": "",
    "data-suppress-scroll-x": "true"
  }, React.createElement("div", null, output.join('')), React.createElement("div", {
    className: "ps__rail-x"
  }, React.createElement("div", {
    className: "ps__thumb-x",
    tabIndex: 0
  })), React.createElement("div", {
    className: "ps__rail-y"
  }, React.createElement("div", {
    className: " ps__thumb-y",
    tabIndex: 0
  })));
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"},{"im":"./views/index","from":"View","name":"View"}];

export let Controller, View;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);
    (require || prop === 'View') && (View = require ? require('./views/index').View : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
