// @bgroup/ui/beyond-toast/beyond-toast.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/react-widgets/page';
import * as dependency_3 from 'react';
import * as dependency_4 from '@bgroup/ui/header';
import * as dependency_5 from '@bgroup/ui/toast';
import * as dependency_6 from '@bgroup/ui/copyable-implementation';
import * as dependency_7 from '@bgroup/ui/form';
import * as dependency_8 from '@bgroup/ui/implementation-result';
import * as dependency_9 from '@beyond-js/kernel/styles';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/beyond-toast"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/react-widgets/page', dependency_2],['react', dependency_3],['@bgroup/ui/header', dependency_4],['@bgroup/ui/toast', dependency_5],['@bgroup/ui/copyable-implementation', dependency_6],['@bgroup/ui/form', dependency_7],['@bgroup/ui/implementation-result', dependency_8],['@beyond-js/kernel/styles', dependency_9]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"beyond-toast","vspecifier":"@bgroup/ui@0.0.34/beyond-toast","is":"page","route":"/components/beyond-toast"}]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-toast')


const ims = new Map();

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 1353635915, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _page = require("@beyond-js/react-widgets/page");
var _views = require("./views");
/*bundle*/class Controller extends _page.PageReactWidgetController {
  get Widget() {
    return _views.Page;
  }
}
exports.Controller = Controller;
}});

/**************************************
INTERNAL MODULE: ./views/implementation
**************************************/

ims.set('./views/implementation', {hash: 2920023839, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.implementationWaring = exports.implementationSuccess = exports.implementationLoading = exports.implementationInfo = exports.implementationError = exports.implementation2 = exports.implementation = void 0;
const implementation = `
import { Toasts, toast } from '@bgroup/ui/toast';
export function View() {
    
    function toggleToast(): void {
         toast.info("This is a toast",);
    }
	return (
		<>
            <Toasts
                position={{
                    bottom: '2rem',
                    right: '2rem',
                }}
            />
            <BeyondButton
                className="btn btn-primary"
                onClick={toggleToast}
            >
                Show error Toast
            </BeyondButton>
        </>
	);
}
`;
exports.implementation = implementation;
const implementation2 = `
import { Toasts, toast } from '@bgroup/ui/toast';
export function View() {
    
    function toggleToast(): void {
         toast.info("This is a toast",);
    }
	return (
		<>
        <Toasts
            position={{
                bottom: '2rem',
                right: '2rem',
            }}
        />
        </>
	);
}
`;
exports.implementation2 = implementation2;
const implementationSuccess = `
function toggleToast(): void {
    toast.success("This is a success toast",);
}
`;
exports.implementationSuccess = implementationSuccess;
const implementationError = `
function toggleToast(): void {
    toast.error("This is a error toast",);
}
`;
exports.implementationError = implementationError;
const implementationWaring = `
function toggleToast(): void {
    toast.waring("This is a waring toast",);
}
`;
exports.implementationWaring = implementationWaring;
const implementationInfo = `
function toggleToast(): void {
    toast.info("This is a info toast",);
}
`;
exports.implementationInfo = implementationInfo;
const implementationLoading = `
function toggleToast(): void {
    toast.loading("This is a loading toast",);
}
`;
exports.implementationLoading = implementationLoading;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 2772416309, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Page = Page;
var React = require("react");
var _view = require("./view");
/*bundle*/function Page() {
  return React.createElement(_view.View, null);
}
}});

/****************************
INTERNAL MODULE: ./views/view
****************************/

ims.set('./views/view', {hash: 1792067090, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.View = View;
var React = require("react");
var _header = require("@bgroup/ui/header");
var _toast = require("@bgroup/ui/toast");
var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
var _implementation = require("./implementation");
var _form = require("@bgroup/ui/form");
var _implementationResult = require("@bgroup/ui/implementation-result");
/*bundle*/function View() {
  function toggleToast(event) {
    const {
      dataset
    } = event.currentTarget;
    _toast.toast[dataset.type](dataset.message);
  }
  return React.createElement("div", null, React.createElement("div", {
    className: "view"
  }, React.createElement(_header.Header, {
    componentName: "Toast"
  }), React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Import:"), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { Toasts, toast } from '@bgroup/ui/toast';`)), React.createElement("div", {
    className: "component-implementation"
  }, React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Implementation:"), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation)), React.createElement("div", {
    className: "component-result"
  }, React.createElement("h3", null, "Result:"), React.createElement(_form.BeyondButton, {
    "data-type": "info",
    "data-message": "This is an example",
    className: "btn btn-primary",
    onClick: toggleToast
  }, "Show Toast"), React.createElement("h3", null, "Case success toast"), React.createElement(_implementationResult.ImplementationResult, {
    code: _implementation.implementationSuccess
  }, React.createElement(_form.BeyondButton, {
    "data-type": "success",
    "data-message": "This is an example of success toast",
    className: "btn btn-primary",
    onClick: toggleToast
  }, "Show success Toast")), React.createElement("h3", null, "Case error toast"), React.createElement(_implementationResult.ImplementationResult, {
    code: _implementation.implementationError
  }, React.createElement(_form.BeyondButton, {
    "data-type": "error",
    "data-message": "This is an example of error toast",
    className: "btn btn-primary",
    onClick: toggleToast
  }, "Show error Toast")), React.createElement("h3", null, "Case warning toast"), React.createElement(_implementationResult.ImplementationResult, {
    code: _implementation.implementationWaring
  }, React.createElement(_form.BeyondButton, {
    "data-type": "warning",
    "data-message": "This is an example of warning toast",
    className: "btn btn-primary",
    onClick: toggleToast
  }, "Show warning Toast")), React.createElement("h3", null, "Case info toast"), React.createElement(_implementationResult.ImplementationResult, {
    code: _implementation.implementationInfo
  }, React.createElement(_form.BeyondButton, {
    "data-type": "info",
    "data-message": "This is an example of info toast",
    className: "btn btn-primary",
    onClick: toggleToast
  }, "Show info Toast")), React.createElement(_implementationResult.ImplementationResult, {
    code: _implementation.implementationLoading
  }, React.createElement(_form.BeyondButton, {
    "data-type": "loading",
    "data-message": "This is an example of loading toast",
    className: "btn btn-primary",
    onClick: toggleToast
  }, "Show loading Toast"))))));
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"},{"im":"./views/index","from":"Page","name":"Page"},{"im":"./views/view","from":"View","name":"View"}];

export let Controller, Page, View;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);
    (require || prop === 'Page') && (Page = require ? require('./views/index').Page : value);
    (require || prop === 'View') && (View = require ? require('./views/view').View : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
