// @bgroup/ui/header/header.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react", "@bgroup/ui/form", "@bgroup/ui/icons"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2, dependency_3, dependency_4) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.Header = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "@bgroup/ui@0.0.34/header"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2], ['@bgroup/ui/form', dependency_3], ['@bgroup/ui/icons', dependency_4]]);
  brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/header');
  const ims = new Map();

  /***********************
  INTERNAL MODULE: ./index
  ***********************/

  ims.set('./index', {
    hash: 3201702456,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Header = Header;
      var React = require("react");
      var _form = require("@bgroup/ui/form");
      var _icons = require("@bgroup/ui/icons");
      ;
      /*bundle*/
      function Header({
        componentName
      }) {
        const theme = globalThis?.matchMedia("(prefers-color-scheme: dark)").matches;
        const themeStorage = localStorage.getItem('theme');
        const defaultTheme = theme ? "dark" : "light";
        const userTheme = themeStorage ? themeStorage === "dark" : defaultTheme === "dark";
        const [isDark, setIsDark] = React.useState(userTheme);
        function handleChange({
          target
        }) {
          const container = document.querySelector("html");
          const theme = isDark ? "light" : "dark";
          container.setAttribute("data-beyond-mode", theme);
          localStorage.setItem("theme", theme);
          setIsDark(target.checked);
        }
        ;
        React.useEffect(() => {
          if (!themeStorage) return;
          const container = document.querySelector("html");
          container.setAttribute("data-beyond-mode", themeStorage);
        }, []);
        const icon = isDark ? "moon" : "sun";
        return React.createElement("header", {
          className: "header"
        }, React.createElement("h2", {
          className: "title"
        }, componentName), React.createElement("div", {
          className: "content-theme"
        }, React.createElement(_icons.BeyondIcon, {
          icon: icon
        }), React.createElement(_form.BeyondSwitch, {
          className: "checkbox",
          checked: isDark,
          onChange: handleChange
        })));
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./index",
    "from": "Header",
    "name": "Header"
  }];
  let Header;

  // Module exports
  _exports.Header = Header;
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'Header') && (_exports.Header = Header = require ? require('./index').Header : value);
  };
  const __beyond_pkg = __pkg;
  _exports.__beyond_pkg = __beyond_pkg;
  const hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  _exports.hmr = hmr;
  __pkg.initialise(ims);
});
