// @bgroup/ui/copyable-implementation/copyable-implementation.browser.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from '@beyond-js/kernel/styles';
import * as dependency_2 from 'react';
import * as dependency_3 from '@bgroup/ui/toast';
import * as dependency_4 from '@bgroup/ui/icons';
import * as dependency_5 from 'prism-react-renderer';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/copyable-implementation"},"type":"code"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1],['react', dependency_2],['@bgroup/ui/toast', dependency_3],['@bgroup/ui/icons', dependency_4],['prism-react-renderer', dependency_5]]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/copyable-implementation')


const ims = new Map();

/***********************
INTERNAL MODULE: ./index
***********************/

ims.set('./index', {hash: 1316914604, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CopyableImplementation = CopyableImplementation;
var React = require("react");
var _toast = require("@bgroup/ui/toast");
var _icons = require("@bgroup/ui/icons");
var _prismReactRenderer = require("prism-react-renderer");
/*bundle*/function CopyableImplementation({
  children
}) {
  function copy() {
    navigator.clipboard.writeText(children);
    _toast.toast.success('Copied implementation');
  }
  return React.createElement("div", {
    className: "copyable-content"
  }, React.createElement("div", {
    className: "code"
  }, React.createElement(_prismReactRenderer.default, {
    ..._prismReactRenderer.defaultProps,
    code: children,
    language: "jsx"
  }, ({
    className,
    style,
    tokens,
    getLineProps,
    getTokenProps
  }) => React.createElement("code", {
    className: className,
    style: style
  }, tokens.map((line, i) => React.createElement("div", {
    ...getLineProps({
      line,
      key: i
    })
  }, line.map((token, key) => React.createElement("span", {
    ...getTokenProps({
      token,
      key
    })
  }))))))), React.createElement(_icons.BeyondIconButton, {
    icon: "copy",
    onClick: copy,
    className: "copy-button"
  }));
}
}});

__pkg.exports.descriptor = [{"im":"./index","from":"CopyableImplementation","name":"CopyableImplementation"}];

export let CopyableImplementation;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'CopyableImplementation') && (CopyableImplementation = require ? require('./index').CopyableImplementation : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
