// @bgroup/ui/input-controller/input-controller.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "react"], function (_exports, _amd_module, dependency_0, dependency_1) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.inputComponents = _exports.hmr = _exports.__beyond_pkg = _exports.Component = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "@bgroup/ui@0.0.34/input-controller"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['react', dependency_1]]);
  const ims = new Map();

  /***********************
  INTERNAL MODULE: ./index
  ***********************/

  ims.set('./index', {
    hash: 3919871860,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Component = Component;
      var React = require("react");
      var _inputComponents = require("./input-components");
      /*bundle*/
      function Component({
        template
      }) {
        const isTemplateAnArray = Array.isArray(template);
        const SingleInput = (!isTemplateAnArray && _inputComponents.inputComponents[template.type]) ?? _inputComponents.inputComponents.default;
        const inputs = isTemplateAnArray && template.map(input => {
          const Element = _inputComponents.inputComponents[input.type] ?? _inputComponents.inputComponents.default;
          return React.createElement(Element, {
            key: input.name,
            ...input,
            _components: _inputComponents.inputComponents
          });
        });
        const output = Array.isArray(template) ? inputs : React.createElement(SingleInput, {
          ...template
        });
        return React.createElement(React.Fragment, null, output);
      }
    }
  });

  /**********************************
  INTERNAL MODULE: ./input-components
  **********************************/

  ims.set('./input-components', {
    hash: 2354356184,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.inputComponents = void 0;
      var _default = require("./inputs/default");
      var _select = require("./inputs/select");
      var _textarea = require("./inputs/textarea");
      var _checkAndRadios = require("./inputs/check-and-radios");
      class InputComponents {
        select = _select.Select;
        textarea = _textarea.TextArea;
        radio = _checkAndRadios.CheckAndRadios;
        radioItem = _default.Input;
        checkbox = _checkAndRadios.CheckAndRadios;
        checkItem = _default.Input;
        default = _default.Input;
        setComponent(type, component) {
          this[type] = component;
        }
      }
      /*bundle*/
      const inputComponents = new InputComponents();
      exports.inputComponents = inputComponents;
    }
  });

  /*****************************************
  INTERNAL MODULE: ./inputs/check-and-radios
  *****************************************/

  ims.set('./inputs/check-and-radios', {
    hash: 3403928382,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.CheckAndRadios = CheckAndRadios;
      var React = require("react");
      function CheckAndRadios(props) {
        const {
          _components
        } = props;
        const output = props.options.map(option => {
          const Element = _components[option.type] ?? _components.default;
          const type = option.type === 'radioItem' ? 'radio' : 'checkbox';
          option = {
            ...option,
            type
          };
          return React.createElement(Element, {
            key: option.value,
            ...option
          });
        });
        return React.createElement(React.Fragment, null, output);
      }
    }
  });

  /********************************
  INTERNAL MODULE: ./inputs/default
  ********************************/

  ims.set('./inputs/default', {
    hash: 1333658770,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Input = Input;
      var React = require("react");
      function Input(props) {
        const {
          label,
          events,
          ...othersSelectProps
        } = props;
        const id = othersSelectProps?.id ?? othersSelectProps.name;
        return React.createElement("label", {
          htmlFor: id
        }, React.createElement("span", null, label), React.createElement("input", {
          ...othersSelectProps,
          ...events,
          id: id
        }));
      }
    }
  });

  /*******************************
  INTERNAL MODULE: ./inputs/select
  *******************************/

  ims.set('./inputs/select', {
    hash: 1243619150,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Select = Select;
      var React = require("react");
      function Select(props) {
        const {
          options,
          label,
          events,
          ...othersSelectProps
        } = props;
        const optionsElements = options.length > 0 && options.map(opt => React.createElement("option", {
          key: opt,
          value: opt
        }, opt));
        const id = othersSelectProps?.id ?? othersSelectProps.name;
        return React.createElement("label", {
          htmlFor: id
        }, React.createElement("span", null, label), React.createElement("select", {
          ...othersSelectProps,
          id: id,
          ...events
        }, optionsElements));
      }
    }
  });

  /*********************************
  INTERNAL MODULE: ./inputs/textarea
  *********************************/

  ims.set('./inputs/textarea', {
    hash: 562440940,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.TextArea = TextArea;
      var React = require("react");
      function TextArea(props) {
        const {
          label,
          events,
          ...othersSelectProps
        } = props;
        const id = othersSelectProps?.id ?? othersSelectProps.name;
        return React.createElement("label", {
          htmlFor: id
        }, React.createElement("span", null, label), React.createElement("textarea", {
          ...othersSelectProps,
          ...events,
          id: id
        }));
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./index",
    "from": "Component",
    "name": "Component"
  }, {
    "im": "./input-components",
    "from": "inputComponents",
    "name": "inputComponents"
  }];
  let Component, inputComponents;

  // Module exports
  _exports.inputComponents = inputComponents;
  _exports.Component = Component;
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'Component') && (_exports.Component = Component = require ? require('./index').Component : value);
    (require || prop === 'inputComponents') && (_exports.inputComponents = inputComponents = require ? require('./input-components').inputComponents : value);
  };
  const __beyond_pkg = __pkg;
  _exports.__beyond_pkg = __beyond_pkg;
  const hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  _exports.hmr = hmr;
  __pkg.initialise(ims);
});
