// @bgroup/ui/beyond-spinner/beyond-spinner.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/react-widgets/page';
import * as dependency_3 from 'react';
import * as dependency_4 from '@bgroup/ui/header';
import * as dependency_5 from '@bgroup/ui/spinner';
import * as dependency_6 from '@bgroup/ui/copyable-implementation';
import * as dependency_7 from '@bgroup/ui/implementation-result';
import * as dependency_8 from '@beyond-js/kernel/styles';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/beyond-spinner"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/react-widgets/page', dependency_2],['react', dependency_3],['@bgroup/ui/header', dependency_4],['@bgroup/ui/spinner', dependency_5],['@bgroup/ui/copyable-implementation', dependency_6],['@bgroup/ui/implementation-result', dependency_7],['@beyond-js/kernel/styles', dependency_8]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"beyond-spinner","vspecifier":"@bgroup/ui@0.0.34/beyond-spinner","is":"page","route":"/components/beyond-spinner"}]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-spinner')


const ims = new Map();

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 3638196390, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _page = require("@beyond-js/react-widgets/page");
var _views = require("./views");
/*bundle*/class Controller extends _page.PageReactWidgetController {
  get Widget() {
    return _views.View;
  }
}
exports.Controller = Controller;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 2332144000, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.View = View;
var React = require("react");
var _header = require("@bgroup/ui/header");
var _spinner = require("@bgroup/ui/spinner");
var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
var _implementationResult = require("@bgroup/ui/implementation-result");
/*bundle*/function View() {
  return React.createElement("div", {
    className: "view"
  }, React.createElement(_header.Header, {
    componentName: "Spinner"
  }), React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Import:"), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { BeyondSpinner } from '@bgroup/ui/spinner';`)), React.createElement("div", {
    className: 'component-implementation'
  }, React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Implementation:"), React.createElement(_copyableImplementation.CopyableImplementation, null, `<BeyondSpinner color="#617096" className="spinner" />`)), React.createElement("div", {
    className: "component-result"
  }, React.createElement("h3", null, "Result:"), React.createElement(_spinner.BeyondSpinner, {
    color: "#617096",
    className: "spinner"
  }), React.createElement("h3", null, "Spinner primary:"), React.createElement(_implementationResult.ImplementationResult, {
    code: `<BeyondSpinner primary className="spinner" />`
  }, React.createElement(_spinner.BeyondSpinner, {
    primary: true,
    className: "spinner"
  })), React.createElement("h3", null, "Spinner secondary:"), React.createElement(_implementationResult.ImplementationResult, {
    code: `<BeyondSpinner secondary className="spinner" />`
  }, React.createElement(_spinner.BeyondSpinner, {
    secondary: true,
    className: "spinner"
  })), React.createElement("h3", null, "Spinner size defined:"), React.createElement(_implementationResult.ImplementationResult, {
    code: `<BeyondSpinner size="80px" className="spinner" />`
  }, React.createElement(_spinner.BeyondSpinner, {
    size: "80px",
    className: "spinner"
  })))));
}
}});

/***************************************
INTERNAL MODULE: ./views/spinner-primary
***************************************/

ims.set('./views/spinner-primary', {hash: 2200184029, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SpinnerPrimary = SpinnerPrimary;
var _react = require("react");
var _spinner = require("@bgroup/ui/spinner");
function SpinnerPrimary() {
  return _react.default.createElement(_spinner.BeyondSpinner, {
    secondary: true,
    className: "spinner"
  });
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"},{"im":"./views/index","from":"View","name":"View"}];

export let Controller, View;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);
    (require || prop === 'View') && (View = require ? require('./views/index').View : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
