// @bgroup/ui/hooks/hooks.browser.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from 'react';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/hooks"},"type":"code"}, import.meta.url).package();;

__pkg.dependencies.update([['react', dependency_1]]);

const ims = new Map();

/****************************
INTERNAL MODULE: ./use-binder
****************************/

ims.set('./use-binder', {hash: 4003813902, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useBinder = useBinder;
var React = require("react");
/*bundle*/ /***
            * Executes a useEffect hook binging the event defined in all
            * objects passed
            *
            * @param {array} objects Objects to bind
            * @param {function} onBinder function to be executed when the event is fired
            * @param {string} event the event to be listened, by default is event change
            */
function useBinder(objects, onBinder, event = 'change') {
  React.useEffect(() => {
    objects.forEach(object => {
      if (!object) {
        throw new Error(`object is not valid in useBinder ${object}`);
      }
      object.bind(event, onBinder);
    });
    return () => objects.forEach(object => object.unbind(event, onBinder));
  }, []);
}
}});

__pkg.exports.descriptor = [{"im":"./use-binder","from":"useBinder","name":"useBinder"}];

export let useBinder;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'useBinder') && (useBinder = require ? require('./use-binder').useBinder : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
