// @bgroup/ui/beyond-alert/beyond-alert.sjs.js
System.register(["@beyond-js/widgets/render", "@beyond-js/kernel/bundle", "@beyond-js/react-widgets/page", "react", "@bgroup/ui/header", "@bgroup/ui/copyable-implementation", "@bgroup/ui/implementation-result", "@bgroup/ui/alert", "@beyond-js/kernel/styles"], function (_export, _context) {
  "use strict";

  var dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5, dependency_6, dependency_7, dependency_8, __Bundle, __pkg, ims, Controller, Page, View, __beyond_pkg, hmr;
  _export({
    Controller: void 0,
    Page: void 0,
    View: void 0
  });
  return {
    setters: [function (_beyondJsWidgetsRender) {
      dependency_0 = _beyondJsWidgetsRender;
    }, function (_beyondJsKernelBundle) {
      dependency_1 = _beyondJsKernelBundle;
    }, function (_beyondJsReactWidgetsPage) {
      dependency_2 = _beyondJsReactWidgetsPage;
    }, function (_react2) {
      dependency_3 = _react2;
    }, function (_bgroupUiHeader) {
      dependency_4 = _bgroupUiHeader;
    }, function (_bgroupUiCopyableImplementation) {
      dependency_5 = _bgroupUiCopyableImplementation;
    }, function (_bgroupUiImplementationResult) {
      dependency_6 = _bgroupUiImplementationResult;
    }, function (_bgroupUiAlert) {
      dependency_7 = _bgroupUiAlert;
    }, function (_beyondJsKernelStyles) {
      dependency_8 = _beyondJsKernelStyles;
    }],
    execute: function () {
      ({
        Bundle: __Bundle
      } = dependency_1);
      __pkg = new __Bundle({
        "module": {
          "vspecifier": "@bgroup/ui@0.0.34/beyond-alert"
        },
        "type": "widget"
      }, _context.meta.url).package();
      ;
      __pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0], ['@beyond-js/react-widgets/page', dependency_2], ['react', dependency_3], ['@bgroup/ui/header', dependency_4], ['@bgroup/ui/copyable-implementation', dependency_5], ['@bgroup/ui/implementation-result', dependency_6], ['@bgroup/ui/alert', dependency_7], ['@beyond-js/kernel/styles', dependency_8]]);
      brequire('@beyond-js/widgets/render').widgets.register([{
        "name": "beyond-alert",
        "vspecifier": "@bgroup/ui@0.0.34/beyond-alert",
        "is": "page",
        "route": "/components/beyond-alert"
      }]);
      brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-alert');
      ims = new Map();
      /****************************
      INTERNAL MODULE: ./controller
      ****************************/
      ims.set('./controller', {
        hash: 1353635915,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Controller = void 0;
          var _page = require("@beyond-js/react-widgets/page");
          var _views = require("./views");
          /*bundle*/
          class Controller extends _page.PageReactWidgetController {
            get Widget() {
              return _views.Page;
            }
          }
          exports.Controller = Controller;
        }
      });

      /*****************************
      INTERNAL MODULE: ./views/cases
      *****************************/

      ims.set('./views/cases', {
        hash: 2514992002,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.cases = void 0;
          const cases = {
            error: {
              message: "this is a error message",
              title: "title error",
              mode: "error",
              show: true,
              isClose: true
            },
            success: {
              message: "this is a success message",
              title: "success error",
              mode: "success",
              show: true,
              isClose: true
            },
            warning: {
              message: "this is a warning message",
              title: "title warning",
              mode: "warning",
              show: true,
              isClose: false
            },
            info: {
              message: "this is a info message",
              title: "title info",
              mode: "info",
              show: true,
              isClose: true
            }
          };
          exports.cases = cases;
        }
      });

      /**************************************
      INTERNAL MODULE: ./views/implementation
      **************************************/

      ims.set('./views/implementation', {
        hash: 1830504499,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.implementationWaring = exports.implementationInfo = exports.implementationError = exports.implementation = void 0;
          const implementation = `
import { useToastContext } from "@bgroup/ui/toast";
export function ImplementationAlert() {
<BeyondAlert  
    message="this is a success message"
    title="title success"
    mode="success"
    show={true}   
    isClose={true}
/>
  }
`;
          exports.implementation = implementation;
          const implementationError = `
<BeyondAlert   
    message="this is a error message"
    title="title error"
    mode="error"
    show={true}
    isClose={true}   
/>
`;
          exports.implementationError = implementationError;
          const implementationWaring = `
<BeyondAlert   
    message="this is a success warning"
    title="title warning"
    mode="warning"
    show={true}   
    isClose={false}
/>
`;
          exports.implementationWaring = implementationWaring;
          const implementationInfo = `
<BeyondAlert   
    message="this is a info message"
    title="title info"
    mode="info"
    show={true} 
    isClose={true}  
/>
`;
          exports.implementationInfo = implementationInfo;
        }
      });

      /*****************************
      INTERNAL MODULE: ./views/index
      *****************************/

      ims.set('./views/index', {
        hash: 2772416309,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Page = Page;
          var React = require("react");
          var _view = require("./view");
          /*bundle*/
          function Page() {
            return React.createElement(_view.View, null);
          }
        }
      });

      /****************************
      INTERNAL MODULE: ./views/view
      ****************************/

      ims.set('./views/view', {
        hash: 78119312,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.View = View;
          var _react = require("react");
          var _header = require("@bgroup/ui/header");
          var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
          var _implementation = require("./implementation");
          var _implementationResult = require("@bgroup/ui/implementation-result");
          var _alert = require("@bgroup/ui/alert");
          var _cases = require("./cases");
          /*bundle*/
          function View() {
            return _react.default.createElement("div", null, _react.default.createElement("div", {
              className: "view"
            }, _react.default.createElement(_header.Header, {
              componentName: "Alert"
            }), _react.default.createElement("div", {
              className: "component-info"
            }, _react.default.createElement("h3", null, "Import:"), _react.default.createElement(_copyableImplementation.CopyableImplementation, null, `import { BeyondAlert } from '@bgroup/ui/alert';`)), _react.default.createElement("div", {
              className: "component-implementation"
            }, _react.default.createElement("div", {
              className: "component-info"
            }, _react.default.createElement("h3", null, "Implementation:"), _react.default.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation)), _react.default.createElement("div", {
              className: "component-result"
            }, _react.default.createElement("h3", null, "Result:"), _react.default.createElement(_alert.BeyondAlert, {
              ..._cases.cases.success
            }), _react.default.createElement("h3", null, "Case success alert"), _react.default.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementationError
            }, _react.default.createElement(_alert.BeyondAlert, {
              ..._cases.cases.error
            })), _react.default.createElement("h3", null, "Case error alert"), _react.default.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementationWaring
            }, _react.default.createElement(_alert.BeyondAlert, {
              ..._cases.cases.warning
            })), _react.default.createElement("h3", null, "Case warning alert"), _react.default.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementationInfo
            }, _react.default.createElement(_alert.BeyondAlert, {
              ..._cases.cases.info
            }))))));
          }
        }
      });
      __pkg.exports.descriptor = [{
        "im": "./controller",
        "from": "Controller",
        "name": "Controller"
      }, {
        "im": "./views/index",
        "from": "Page",
        "name": "Page"
      }, {
        "im": "./views/view",
        "from": "View",
        "name": "View"
      }];
      // Module exports
      __pkg.exports.process = function ({
        require,
        prop,
        value
      }) {
        (require || prop === 'Controller') && _export("Controller", Controller = require ? require('./controller').Controller : value);
        (require || prop === 'Page') && _export("Page", Page = require ? require('./views/index').Page : value);
        (require || prop === 'View') && _export("View", View = require ? require('./views/view').View : value);
      };
      _export("__beyond_pkg", __beyond_pkg = __pkg);
      _export("hmr", hmr = new function () {
        this.on = (event, listener) => void 0;
        this.off = (event, listener) => void 0;
      }());
      __pkg.initialise(ims);
    }
  };
});
