// @bgroup/ui/beyond-button/beyond-button.sjs.js
System.register(["@beyond-js/widgets/render", "@beyond-js/kernel/bundle", "@beyond-js/react-widgets/page", "react", "@bgroup/ui/form", "@bgroup/ui/header", "@bgroup/ui/copyable-implementation", "@bgroup/ui/implementation-result", "@beyond-js/kernel/styles"], function (_export, _context) {
  "use strict";

  var dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5, dependency_6, dependency_7, dependency_8, __Bundle, __pkg, ims, Controller, View, __beyond_pkg, hmr;
  _export({
    Controller: void 0,
    View: void 0
  });
  return {
    setters: [function (_beyondJsWidgetsRender) {
      dependency_0 = _beyondJsWidgetsRender;
    }, function (_beyondJsKernelBundle) {
      dependency_1 = _beyondJsKernelBundle;
    }, function (_beyondJsReactWidgetsPage) {
      dependency_2 = _beyondJsReactWidgetsPage;
    }, function (_react2) {
      dependency_3 = _react2;
    }, function (_bgroupUiForm) {
      dependency_4 = _bgroupUiForm;
    }, function (_bgroupUiHeader) {
      dependency_5 = _bgroupUiHeader;
    }, function (_bgroupUiCopyableImplementation) {
      dependency_6 = _bgroupUiCopyableImplementation;
    }, function (_bgroupUiImplementationResult) {
      dependency_7 = _bgroupUiImplementationResult;
    }, function (_beyondJsKernelStyles) {
      dependency_8 = _beyondJsKernelStyles;
    }],
    execute: function () {
      ({
        Bundle: __Bundle
      } = dependency_1);
      __pkg = new __Bundle({
        "module": {
          "vspecifier": "@bgroup/ui@0.0.34/beyond-button"
        },
        "type": "widget"
      }, _context.meta.url).package();
      ;
      __pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0], ['@beyond-js/react-widgets/page', dependency_2], ['react', dependency_3], ['@bgroup/ui/form', dependency_4], ['@bgroup/ui/header', dependency_5], ['@bgroup/ui/copyable-implementation', dependency_6], ['@bgroup/ui/implementation-result', dependency_7], ['@beyond-js/kernel/styles', dependency_8]]);
      brequire('@beyond-js/widgets/render').widgets.register([{
        "name": "beyond-widget",
        "vspecifier": "@bgroup/ui@0.0.34/beyond-button",
        "is": "page",
        "route": "/components/beyond-button"
      }]);
      brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-button');
      ims = new Map();
      /****************************
      INTERNAL MODULE: ./controller
      ****************************/
      ims.set('./controller', {
        hash: 3638196390,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Controller = void 0;
          var _page = require("@beyond-js/react-widgets/page");
          var _views = require("./views");
          /*bundle*/
          class Controller extends _page.PageReactWidgetController {
            get Widget() {
              return _views.View;
            }
          }
          exports.Controller = Controller;
        }
      });

      /*************************************
      INTERNAL MODULE: ./views/button-border
      *************************************/

      ims.set('./views/button-border', {
        hash: 1695944781,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultsBorder = ResultsBorder;
          var _react = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultsBorder() {
            return _react.default.createElement(_form.BeyondButton, {
              className: "btn btn-border"
            }, "Beyond Button border");
          }
        }
      });

      /*************************************
      INTERNAL MODULE: ./views/button-cancel
      *************************************/

      ims.set('./views/button-cancel', {
        hash: 3958965513,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultsCancel = ResultsCancel;
          var _react = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultsCancel() {
            return _react.default.createElement(_form.BeyondButton, {
              className: "btn btn-cancel"
            }, "Beyond Button cancel");
          }
        }
      });

      /****************************************
      INTERNAL MODULE: ./views/button-secondary
      ****************************************/

      ims.set('./views/button-secondary', {
        hash: 2576662872,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultsSecondary = ResultsSecondary;
          var _react = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultsSecondary() {
            return _react.default.createElement(_form.BeyondButton, {
              className: "btn btn-secondary"
            }, "Beyond Button secondary");
          }
        }
      });

      /**************************************
      INTERNAL MODULE: ./views/implementation
      **************************************/

      ims.set('./views/implementation', {
        hash: 2750882032,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.implementationSpinner = exports.implementationSecondary = exports.implementationIcon = exports.implementationCancel = exports.implementationBorder = exports.implementation = void 0;
          const implementation = `
export function ButtonImplementation() {

	function handleChange(event) {
		event.stopPropagation();
		// logic...
	};
	
	return (
		<BeyondButton className="btn btn-primary" onClick={handleChange}>Beyond Button</BeyondButton>
	);
}`;
          exports.implementation = implementation;
          const implementationSpinner = `
<BeyondButton
	className="btn btn-primary"
	loading={true}
	colorSpinner="#fff"
	>
	Beyond Button with spinner
</BeyondButton>
}`;
          exports.implementationSpinner = implementationSpinner;
          const implementationIcon = `
<BeyondButton
	className="btn btn-primary"
	icon="delete"
	>
	Button icon
</BeyondButton>
}`;
          exports.implementationIcon = implementationIcon;
          const implementationSecondary = `
<BeyondButton className="btn btn-secondary">
	Button secondary
</BeyondButton>
}`;
          exports.implementationSecondary = implementationSecondary;
          const implementationBorder = `
<BeyondButton className="btn btn-border">
	Button border
</BeyondButton>
}`;
          exports.implementationBorder = implementationBorder;
          const implementationCancel = `
<BeyondButton className="btn btn-cancel">
	Button cancel
</BeyondButton>
}`;
          exports.implementationCancel = implementationCancel;
        }
      });

      /*****************************
      INTERNAL MODULE: ./views/index
      *****************************/

      ims.set('./views/index', {
        hash: 2466793613,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.View = View;
          var React = require("react");
          var _header = require("@bgroup/ui/header");
          var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
          var _results = require("./results");
          var _implementation = require("./implementation");
          var _implementationResult = require("@bgroup/ui/implementation-result");
          var _resultSipnner = require("./result-sipnner");
          var _resultIcon = require("./result-icon");
          var _buttonSecondary = require("./button-secondary");
          var _buttonBorder = require("./button-border");
          var _buttonCancel = require("./button-cancel");
          /*bundle*/
          function View() {
            return React.createElement("div", {
              className: "view"
            }, React.createElement(_header.Header, {
              componentName: "Checkbox"
            }), React.createElement("div", {
              className: "component-info"
            }, React.createElement("h3", null, "Import: "), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { BeyondButton } from "@bgroup/ui/form";`)), React.createElement("div", {
              className: 'component-implementation'
            }, React.createElement("div", {
              className: "component-info"
            }, React.createElement("h3", null, "Implementation:"), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation)), React.createElement("div", {
              className: "component-result"
            }, React.createElement("h3", null, "Result button primary:"), React.createElement(_results.Results, null), React.createElement("h3", null, "Case button secondary:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementationSecondary
            }, React.createElement(_buttonSecondary.ResultsSecondary, null)), React.createElement("h3", null, "Case button border:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementationBorder
            }, React.createElement(_buttonBorder.ResultsBorder, null)), React.createElement("h3", null, "Case button cancel:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementationCancel
            }, React.createElement(_buttonCancel.ResultsCancel, null)), React.createElement("h3", null, "Case button loading:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementationSpinner
            }, React.createElement(_resultSipnner.ResultsSpinner, null)), React.createElement("h3", null, "Case button with icon:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementation.implementationIcon
            }, React.createElement(_resultIcon.ResultIcon, null)))));
          }
        }
      });

      /***********************************
      INTERNAL MODULE: ./views/result-icon
      ***********************************/

      ims.set('./views/result-icon', {
        hash: 2717346830,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultIcon = ResultIcon;
          var _react = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultIcon() {
            function handleChange(event) {
              event.stopPropagation();
              // logic...
            }

            return _react.default.createElement(_form.BeyondButton, {
              className: "btn btn-primary",
              icon: "delete",
              onClick: handleChange
            }, "Button with icon");
          }
        }
      });

      /**************************************
      INTERNAL MODULE: ./views/result-sipnner
      **************************************/

      ims.set('./views/result-sipnner', {
        hash: 702096365,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultsSpinner = ResultsSpinner;
          var _react = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultsSpinner() {
            function handleChange(event) {
              event.stopPropagation();
              // logic...
            }

            return _react.default.createElement(_form.BeyondButton, {
              className: "btn btn-primary spinner",
              loading: true,
              colorSpinner: "#fff",
              onClick: handleChange
            }, "Beyond Button with spinner");
          }
        }
      });

      /*******************************
      INTERNAL MODULE: ./views/results
      *******************************/

      ims.set('./views/results', {
        hash: 120515186,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Results = Results;
          var _react = require("react");
          var _form = require("@bgroup/ui/form");
          function Results() {
            function handleChange(event) {
              event.stopPropagation();
              // logic...
            }

            ;
            return _react.default.createElement(_form.BeyondButton, {
              className: "btn btn-primary",
              onClick: handleChange
            }, "Beyond Button");
          }
        }
      });
      __pkg.exports.descriptor = [{
        "im": "./controller",
        "from": "Controller",
        "name": "Controller"
      }, {
        "im": "./views/index",
        "from": "View",
        "name": "View"
      }];
      // Module exports
      __pkg.exports.process = function ({
        require,
        prop,
        value
      }) {
        (require || prop === 'Controller') && _export("Controller", Controller = require ? require('./controller').Controller : value);
        (require || prop === 'View') && _export("View", View = require ? require('./views/index').View : value);
      };
      _export("__beyond_pkg", __beyond_pkg = __pkg);
      _export("hmr", hmr = new function () {
        this.on = (event, listener) => void 0;
        this.off = (event, listener) => void 0;
      }());
      __pkg.initialise(ims);
    }
  };
});
