// @bgroup/ui/beyond-empty/beyond-empty.amd.js
define(["exports", "module", "@beyond-js/widgets/render", "@beyond-js/kernel/bundle", "@beyond-js/react-widgets/page", "react", "@bgroup/ui/header", "@bgroup/ui/empty", "@bgroup/ui/copyable-implementation", "@beyond-js/kernel/styles"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5, dependency_6, dependency_7) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.View = _exports.Controller = void 0;
  const {
    Bundle: __Bundle
  } = dependency_1;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "@bgroup/ui@0.0.34/beyond-empty"
    },
    "type": "widget"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0], ['@beyond-js/react-widgets/page', dependency_2], ['react', dependency_3], ['@bgroup/ui/header', dependency_4], ['@bgroup/ui/empty', dependency_5], ['@bgroup/ui/copyable-implementation', dependency_6], ['@beyond-js/kernel/styles', dependency_7]]);
  brequire('@beyond-js/widgets/render').widgets.register([{
    "name": "beyond-empty",
    "vspecifier": "@bgroup/ui@0.0.34/beyond-empty",
    "is": "page",
    "route": "/components/beyond-empty"
  }]);
  brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-empty');
  const ims = new Map();

  /****************************
  INTERNAL MODULE: ./controller
  ****************************/

  ims.set('./controller', {
    hash: 3638196390,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Controller = void 0;
      var _page = require("@beyond-js/react-widgets/page");
      var _views = require("./views");
      /*bundle*/
      class Controller extends _page.PageReactWidgetController {
        get Widget() {
          return _views.View;
        }
      }
      exports.Controller = Controller;
    }
  });

  /**************************************
  INTERNAL MODULE: ./views/implementation
  **************************************/

  ims.set('./views/implementation', {
    hash: 1445532229,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.implementation = void 0;
      const implementation = `
<BeyondEmpty text="No records to display" icon="info" />
`;
      exports.implementation = implementation;
    }
  });

  /*****************************
  INTERNAL MODULE: ./views/index
  *****************************/

  ims.set('./views/index', {
    hash: 498391636,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.View = View;
      var React = require("react");
      var _header = require("@bgroup/ui/header");
      var _empty = require("@bgroup/ui/empty");
      var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
      var _implementation = require("./implementation");
      /*bundle*/
      function View() {
        return React.createElement("div", {
          className: "view"
        }, React.createElement(_header.Header, {
          componentName: "Empty"
        }), React.createElement("div", {
          className: "component-info"
        }, React.createElement("h3", null, "Import: "), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { BeyondEmpty } from '@bgroup/ui/empty';`)), React.createElement("div", {
          className: 'component-implementation'
        }, React.createElement("div", {
          className: "component-info"
        }, React.createElement("h3", null, "Implementation: "), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation)), React.createElement("div", {
          className: "component-result"
        }, React.createElement("h3", null, "Result:"), React.createElement(_empty.BeyondEmpty, {
          icon: "info",
          text: "No records to display"
        }))));
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./controller",
    "from": "Controller",
    "name": "Controller"
  }, {
    "im": "./views/index",
    "from": "View",
    "name": "View"
  }];
  let Controller, View;

  // Module exports
  _exports.View = View;
  _exports.Controller = Controller;
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'Controller') && (_exports.Controller = Controller = require ? require('./controller').Controller : value);
    (require || prop === 'View') && (_exports.View = View = require ? require('./views/index').View : value);
  };
  const __beyond_pkg = __pkg;
  _exports.__beyond_pkg = __beyond_pkg;
  const hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  _exports.hmr = hmr;
  __pkg.initialise(ims);
});
