// @bgroup/ui/alert/alert.sjs.js
System.register(["@beyond-js/kernel/bundle", "react", "@bgroup/ui/icons", "@beyond-js/kernel/styles"], function (_export, _context) {
  "use strict";

  var dependency_0, dependency_1, dependency_2, dependency_3, __Bundle, __pkg, ims, BeyondAlert, __beyond_pkg, hmr;
  _export("BeyondAlert", void 0);
  return {
    setters: [function (_beyondJsKernelBundle) {
      dependency_0 = _beyondJsKernelBundle;
    }, function (_react2) {
      dependency_1 = _react2;
    }, function (_bgroupUiIcons) {
      dependency_2 = _bgroupUiIcons;
    }, function (_beyondJsKernelStyles) {
      dependency_3 = _beyondJsKernelStyles;
    }],
    execute: function () {
      ({
        Bundle: __Bundle
      } = dependency_0);
      __pkg = new __Bundle({
        "module": {
          "vspecifier": "@bgroup/ui@0.0.34/alert"
        },
        "type": "code",
        "name": "alert"
      }, _context.meta.url).package();
      ;
      __pkg.dependencies.update([['react', dependency_1], ['@bgroup/ui/icons', dependency_2], ['@beyond-js/kernel/styles', dependency_3]]);
      brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/alert');
      ims = new Map();
      /***********************
      INTERNAL MODULE: ./alert
      ***********************/
      ims.set('./alert', {
        hash: 2419189041,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.BeyondAlert = BeyondAlert;
          var _react = require("react");
          var _icons = require("@bgroup/ui/icons");
          /*bundle*/
          function BeyondAlert(props) {
            const {
              message,
              show,
              className,
              mode,
              title,
              isClose
            } = props;
            const [state, setState] = _react.default.useState({
              show: show ?? false,
              text: message ?? "texto de la alerta"
            });
            _react.default.useEffect(() => {
              setState({
                ...state,
                show: show
              });
            }, [show]);
            if (!state.show) return null;
            const close = () => setState({
              ...state,
              show: false,
              text: ""
            });
            const cls = className ? `${className} alert ${mode ?? "success"}` : `alert ${mode ?? "success"}`;
            return _react.default.createElement("div", {
              className: cls,
              onClick: close
            }, " ", title && _react.default.createElement("span", {
              dangerouslySetInnerHTML: {
                __html: title
              }
            }), message && _react.default.createElement("span", {
              dangerouslySetInnerHTML: {
                __html: message
              }
            }), isClose && _react.default.createElement(_icons.BeyondIconButton, {
              icon: "close",
              onClick: close
            }));
          }
        }
      });
      __pkg.exports.descriptor = [{
        "im": "./alert",
        "from": "BeyondAlert",
        "name": "BeyondAlert"
      }];
      // Module exports
      __pkg.exports.process = function ({
        require,
        prop,
        value
      }) {
        (require || prop === 'BeyondAlert') && _export("BeyondAlert", BeyondAlert = require ? require('./alert').BeyondAlert : value);
      };
      _export("__beyond_pkg", __beyond_pkg = __pkg);
      _export("hmr", hmr = new function () {
        this.on = (event, listener) => void 0;
        this.off = (event, listener) => void 0;
      }());
      __pkg.initialise(ims);
    }
  };
});
