// @bgroup/ui/beyond-switch/beyond-switch.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/react-widgets/page';
import * as dependency_3 from 'react';
import * as dependency_4 from '@bgroup/ui/header';
import * as dependency_5 from '@bgroup/ui/copyable-implementation';
import * as dependency_6 from '@bgroup/ui/form';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/beyond-switch"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/react-widgets/page', dependency_2],['react', dependency_3],['@bgroup/ui/header', dependency_4],['@bgroup/ui/copyable-implementation', dependency_5],['@bgroup/ui/form', dependency_6]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"beyond-switch","vspecifier":"@bgroup/ui@0.0.34/beyond-switch","is":"page","route":"/components/beyond-switch"}]);

const ims = new Map();

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 3638196390, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _page = require("@beyond-js/react-widgets/page");
var _views = require("./views");
/*bundle*/class Controller extends _page.PageReactWidgetController {
  get Widget() {
    return _views.View;
  }
}
exports.Controller = Controller;
}});

/**************************************
INTERNAL MODULE: ./views/implementation
**************************************/

ims.set('./views/implementation', {hash: 4127072298, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.implementation = void 0;
const implementation = `
export function SwitchImplementation() {
	const [isActive, setIsActive] = React.useState(true);

	function handleChange() {
		setIsActive(!isActive);
	}

	function handleSubmit(event) {
		event.preventDefault();
		// Logic...
	}

	return (
		<form onSubmit={handleSubmit} className="results-form">
			<BeyondSwitch className="checkbox" checked={isActive} onChange={handleChange} />
		</form>
	);
}`;
exports.implementation = implementation;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 3161907421, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.View = View;
var React = require("react");
var _header = require("@bgroup/ui/header");
var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
var _results = require("./results");
var _implementation = require("./implementation");
/*bundle*/function View() {
  return React.createElement("div", {
    className: "view"
  }, React.createElement(_header.Header, {
    componentName: "Switch"
  }), React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Import:"), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { BeyondSwitch } from '@bgroup/ui/form';`)), React.createElement("div", {
    className: 'component-implementation'
  }, React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Implementation: "), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation)), React.createElement("div", {
    className: "component-result"
  }, React.createElement("h3", null, "Result:"), React.createElement(_results.Results, null))));
}
}});

/*******************************
INTERNAL MODULE: ./views/results
*******************************/

ims.set('./views/results', {hash: 2009803870, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Results = Results;
var React = require("react");
var _form = require("@bgroup/ui/form");
function Results() {
  const [isActive, setIsActive] = React.useState(true);
  function handleChange() {
    setIsActive(!isActive);
  }
  function handleSubmit(event) {
    event.preventDefault();
    // Logic...
  }

  return React.createElement("form", {
    onSubmit: handleSubmit,
    className: "results-form"
  }, React.createElement(_form.BeyondSwitch, {
    className: "checkbox",
    checked: isActive,
    onChange: handleChange
  }));
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"},{"im":"./views/index","from":"View","name":"View"}];

export let Controller, View;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);
    (require || prop === 'View') && (View = require ? require('./views/index').View : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
