// @bgroup/ui/beyond-tabs/beyond-tabs.sjs.js
System.register(["@beyond-js/widgets/render", "@beyond-js/kernel/bundle", "@beyond-js/react-widgets/page", "react", "@bgroup/ui/header", "@bgroup/ui/copyable-implementation", "@bgroup/ui/tabs", "@beyond-js/kernel/styles"], function (_export, _context) {
  "use strict";

  var dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5, dependency_6, dependency_7, __Bundle, __pkg, ims, Controller, View, __beyond_pkg, hmr;
  _export({
    Controller: void 0,
    View: void 0
  });
  return {
    setters: [function (_beyondJsWidgetsRender) {
      dependency_0 = _beyondJsWidgetsRender;
    }, function (_beyondJsKernelBundle) {
      dependency_1 = _beyondJsKernelBundle;
    }, function (_beyondJsReactWidgetsPage) {
      dependency_2 = _beyondJsReactWidgetsPage;
    }, function (_react) {
      dependency_3 = _react;
    }, function (_bgroupUiHeader) {
      dependency_4 = _bgroupUiHeader;
    }, function (_bgroupUiCopyableImplementation) {
      dependency_5 = _bgroupUiCopyableImplementation;
    }, function (_bgroupUiTabs) {
      dependency_6 = _bgroupUiTabs;
    }, function (_beyondJsKernelStyles) {
      dependency_7 = _beyondJsKernelStyles;
    }],
    execute: function () {
      ({
        Bundle: __Bundle
      } = dependency_1);
      __pkg = new __Bundle({
        "module": {
          "vspecifier": "@bgroup/ui@0.0.34/beyond-tabs"
        },
        "type": "widget"
      }, _context.meta.url).package();
      ;
      __pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0], ['@beyond-js/react-widgets/page', dependency_2], ['react', dependency_3], ['@bgroup/ui/header', dependency_4], ['@bgroup/ui/copyable-implementation', dependency_5], ['@bgroup/ui/tabs', dependency_6], ['@beyond-js/kernel/styles', dependency_7]]);
      brequire('@beyond-js/widgets/render').widgets.register([{
        "name": "beyond-tabs",
        "vspecifier": "@bgroup/ui@0.0.34/beyond-tabs",
        "is": "page",
        "route": "/components/beyond-tabs"
      }]);
      brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-tabs');
      ims = new Map();
      /****************************
      INTERNAL MODULE: ./controller
      ****************************/
      ims.set('./controller', {
        hash: 784911761,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Controller = void 0;
          var _page = require("@beyond-js/react-widgets/page");
          var _views = require("./views");
          /*bundle*/
          class Controller extends _page.PageReactWidgetController {
            get Widget() {
              return _views.View;
            }
          }
          exports.Controller = Controller;
          ;
        }
      });

      /**************************************
      INTERNAL MODULE: ./views/implementation
      **************************************/

      ims.set('./views/implementation', {
        hash: 2950274872,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.implementation = void 0;
          const implementation = `
export function ImplementationTabs() {
    const [tab, setTab] = React.useState<number>(0);

	return (
        <BeyondTabs selected={tab}>
        <Tabs selected={tab}>
            <button data-path="/components/beyond-tabs?tab=0" onClick={() => setTab(0)}>TAB 0</button>
            <button data-path="/components/beyond-tabs?tab=1" onClick={() => setTab(1)}>TAB 1</button>
            <button data-path="/components/beyond-tabs?tab=2" onClick={() => setTab(2)}>TAB 2</button>
            <button data-path="/components/beyond-tabs?tab=3" onClick={() => setTab(3)}>TAB 3</button>
        </Tabs>
        <TabsContent>
            <div><p>Tab Content 0</p></div>
            <div><p>Tab Content 1</p></div>
            <div><p>Tab Content 2</p></div>
            <div><p>Tab Content 3</p></div>
        </TabsContent>
    </BeyondTabs>
	);
  }
`;
          exports.implementation = implementation;
        }
      });

      /*****************************
      INTERNAL MODULE: ./views/index
      *****************************/

      ims.set('./views/index', {
        hash: 2555593060,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.View = View;
          var React = require("react");
          var _header = require("@bgroup/ui/header");
          var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
          var _implementation = require("./implementation");
          var _results = require("./results");
          /*bundle*/
          function View() {
            return React.createElement("div", {
              className: "view"
            }, React.createElement(_header.Header, {
              componentName: "Tabs"
            }), React.createElement("div", {
              className: "component-info"
            }, React.createElement("h3", null, "Import:"), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { TabsContent, BeyondTabs, Tabs } from "@bgroup/ui/tabs";`)), React.createElement("div", {
              className: "component-implementation"
            }, React.createElement("div", {
              className: "component-info"
            }, React.createElement("h3", null, "Implementation: "), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation), React.createElement("div", {
              className: "component-detail"
            }, React.createElement("h4", null, "Additional Information:"), React.createElement("div", null, React.createElement("p", null, "If you pass the", React.createElement("code", null, "data-path"), " property over the tabs each tab will do a", React.createElement("code", null, "routing.replaceState"), " on the value passed to data-path. tab will do a routing.replaceState on the value passed to ", React.createElement("code", null, "data-path"))))), React.createElement("div", {
              className: "component-result"
            }, React.createElement("h3", null, "Result:"), React.createElement(_results.Results, null))));
          }
        }
      });

      /*******************************
      INTERNAL MODULE: ./views/results
      *******************************/

      ims.set('./views/results', {
        hash: 3266356388,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Results = Results;
          var React = require("react");
          var _tabs = require("@bgroup/ui/tabs");
          function Results() {
            const [tab, setTab] = React.useState(0);
            return React.createElement(_tabs.BeyondTabs, {
              selected: tab
            }, React.createElement(_tabs.Tabs, {
              selected: tab
            }, React.createElement("button", {
              "data-path": "/components/beyond-tabs?tab=0",
              onClick: () => setTab(0)
            }, "TAB 0"), React.createElement("button", {
              "data-path": "/components/beyond-tabs?tab=1",
              onClick: () => setTab(1)
            }, "TAB 1"), React.createElement("button", {
              "data-path": "/components/beyond-tabs?tab=2",
              onClick: () => setTab(2)
            }, "TAB 2"), React.createElement("button", {
              "data-path": "/components/beyond-tabs?tab=3",
              onClick: () => setTab(3)
            }, "TAB 3")), React.createElement(_tabs.TabsContent, null, React.createElement("div", null, React.createElement("p", null, "Tab Content 0")), React.createElement("div", null, React.createElement("p", null, "Tab Content 1")), React.createElement("div", null, React.createElement("p", null, "Tab Content 2")), React.createElement("div", null, React.createElement("p", null, "Tab Content 3"))));
          }
        }
      });
      __pkg.exports.descriptor = [{
        "im": "./controller",
        "from": "Controller",
        "name": "Controller"
      }, {
        "im": "./views/index",
        "from": "View",
        "name": "View"
      }];
      // Module exports
      __pkg.exports.process = function ({
        require,
        prop,
        value
      }) {
        (require || prop === 'Controller') && _export("Controller", Controller = require ? require('./controller').Controller : value);
        (require || prop === 'View') && _export("View", View = require ? require('./views/index').View : value);
      };
      _export("__beyond_pkg", __beyond_pkg = __pkg);
      _export("hmr", hmr = new function () {
        this.on = (event, listener) => void 0;
        this.off = (event, listener) => void 0;
      }());
      __pkg.initialise(ims);
    }
  };
});
