// @bgroup/ui/beyond-input/beyond-input.sjs.js
System.register(["@beyond-js/widgets/render", "@beyond-js/kernel/bundle", "@beyond-js/react-widgets/page", "react", "@bgroup/ui/header", "@bgroup/ui/copyable-implementation", "@bgroup/ui/implementation-result", "@bgroup/ui/form", "@beyond-js/kernel/styles"], function (_export, _context) {
  "use strict";

  var dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5, dependency_6, dependency_7, dependency_8, __Bundle, __pkg, ims, Controller, View, __beyond_pkg, hmr;
  _export({
    Controller: void 0,
    View: void 0
  });
  return {
    setters: [function (_beyondJsWidgetsRender) {
      dependency_0 = _beyondJsWidgetsRender;
    }, function (_beyondJsKernelBundle) {
      dependency_1 = _beyondJsKernelBundle;
    }, function (_beyondJsReactWidgetsPage) {
      dependency_2 = _beyondJsReactWidgetsPage;
    }, function (_react) {
      dependency_3 = _react;
    }, function (_bgroupUiHeader) {
      dependency_4 = _bgroupUiHeader;
    }, function (_bgroupUiCopyableImplementation) {
      dependency_5 = _bgroupUiCopyableImplementation;
    }, function (_bgroupUiImplementationResult) {
      dependency_6 = _bgroupUiImplementationResult;
    }, function (_bgroupUiForm) {
      dependency_7 = _bgroupUiForm;
    }, function (_beyondJsKernelStyles) {
      dependency_8 = _beyondJsKernelStyles;
    }],
    execute: function () {
      ({
        Bundle: __Bundle
      } = dependency_1);
      __pkg = new __Bundle({
        "module": {
          "vspecifier": "@bgroup/ui@0.0.34/beyond-input"
        },
        "type": "widget"
      }, _context.meta.url).package();
      ;
      __pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0], ['@beyond-js/react-widgets/page', dependency_2], ['react', dependency_3], ['@bgroup/ui/header', dependency_4], ['@bgroup/ui/copyable-implementation', dependency_5], ['@bgroup/ui/implementation-result', dependency_6], ['@bgroup/ui/form', dependency_7], ['@beyond-js/kernel/styles', dependency_8]]);
      brequire('@beyond-js/widgets/render').widgets.register([{
        "name": "beyond-input",
        "vspecifier": "@bgroup/ui@0.0.34/beyond-input",
        "is": "page",
        "route": "/components/beyond-input"
      }]);
      brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-input');
      ims = new Map();
      /****************************
      INTERNAL MODULE: ./controller
      ****************************/
      ims.set('./controller', {
        hash: 3638196390,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Controller = void 0;
          var _page = require("@beyond-js/react-widgets/page");
          var _views = require("./views");
          /*bundle*/
          class Controller extends _page.PageReactWidgetController {
            get Widget() {
              return _views.View;
            }
          }
          exports.Controller = Controller;
        }
      });

      /***************************************
      INTERNAL MODULE: ./views/implementations
      ***************************************/

      ims.set('./views/implementations', {
        hash: 1216050490,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.implementationPassword = exports.implementationLoading = exports.implementationIcon = exports.implementationError = exports.implementationDisabled = exports.implementation = void 0;
          const implementation = `
export function Results() {
	const [value, setValue] = React.useState('');

	function handleChange(event) {
		const { value: inputValue } = event.target;
		setValue(inputValue);
	}

	function handleSubmit(event) {
		event.preventDefault();
		// logic...
	}

	return (
		<form onSubmit={handleSubmit}>
			<BeyondInput type="text" name="name" label="Insert a text..." onChange={handleChange} value={value} />
		</form>
	);
}`;
          exports.implementation = implementation;
          const implementationError = `
<BeyondInput
	type="text"
	label="Insert a text..."
	hasError={true}
	errorMessage="Error in input"
/>
`;
          exports.implementationError = implementationError;
          const implementationIcon = `
<BeyondInput
	type="text"
	label="Insert a text..."
	icon="user"
/>
`;
          exports.implementationIcon = implementationIcon;
          const implementationDisabled = `
<BeyondInput
	type="text"
	label="Insert a text..."
	disabled
/>
`;
          exports.implementationDisabled = implementationDisabled;
          const implementationPassword = `
<BeyondInput
	type="password"
	label="Insert a text..."
	password
/>
`;
          exports.implementationPassword = implementationPassword;
          const implementationLoading = `
<BeyondInput
	type="text"
	label="Insert a text..."
	loading
	colorSpinner="#fff"
/>
`;
          exports.implementationLoading = implementationLoading;
        }
      });

      /*****************************
      INTERNAL MODULE: ./views/index
      *****************************/

      ims.set('./views/index', {
        hash: 1212958359,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.View = View;
          var React = require("react");
          var _header = require("@bgroup/ui/header");
          var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
          var _implementations = require("./implementations");
          var _results = require("./results");
          var _resultInputError = require("./result-input-error");
          var _resultInputWithIcon = require("./result-input-with-icon");
          var _resultInputDisabled = require("./result-input-disabled");
          var _resultInputPassword = require("./result-input-password");
          var _resultInputLoading = require("./result-input-loading");
          var _implementationResult = require("@bgroup/ui/implementation-result");
          /*bundle*/
          function View() {
            return React.createElement("div", {
              className: "view"
            }, React.createElement(_header.Header, {
              componentName: "Input"
            }), React.createElement("div", {
              className: "component-info"
            }, React.createElement("h3", null, "Import:"), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { BeyondInput } from '@bgroup/ui/form';`)), React.createElement("div", {
              className: "component-implementation"
            }, React.createElement("div", {
              className: "component-info"
            }, React.createElement("h3", null, "Implementation:"), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementations.implementation)), React.createElement("div", {
              className: "component-result"
            }, React.createElement("h3", null, "Result:"), React.createElement(_results.Results, null), React.createElement("h3", null, "Case input error:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementations.implementationError
            }, React.createElement(_resultInputError.ResultInputError, null)), React.createElement("h3", null, "Case input with icon:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementations.implementationIcon
            }, React.createElement(_resultInputWithIcon.ResultInputWithIcon, null)), React.createElement("h3", null, "Case input disabled:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementations.implementationDisabled
            }, React.createElement(_resultInputDisabled.ResultInputDisabled, null)), React.createElement("h3", null, "Case input password:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementations.implementationPassword
            }, React.createElement(_resultInputPassword.ResultInputPassword, null)), React.createElement("h3", null, "Case input loading:"), React.createElement(_implementationResult.ImplementationResult, {
              code: _implementations.implementationLoading
            }, React.createElement(_resultInputLoading.ResultInputLoading, null)))));
          }
        }
      });

      /*********************************************
      INTERNAL MODULE: ./views/result-input-disabled
      *********************************************/

      ims.set('./views/result-input-disabled', {
        hash: 2616881785,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultInputDisabled = ResultInputDisabled;
          var React = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultInputDisabled() {
            const [value, setValue] = React.useState("");
            function handleChange(event) {
              const {
                value: inputValue
              } = event.target;
              setValue(inputValue);
            }
            ;
            function handleSubmit(event) {
              event.preventDefault();
              // logic...
            }

            ;
            return React.createElement("form", {
              onSubmit: handleSubmit
            }, React.createElement(_form.BeyondInput, {
              type: "text",
              name: "name-disabled",
              label: "Insert a text...",
              onChange: handleChange,
              value: value,
              disabled: true
            }));
          }
        }
      });

      /******************************************
      INTERNAL MODULE: ./views/result-input-error
      ******************************************/

      ims.set('./views/result-input-error', {
        hash: 2894186519,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultInputError = ResultInputError;
          var React = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultInputError() {
            const [value, setValue] = React.useState("");
            function handleChange(event) {
              const {
                value: inputValue
              } = event.target;
              setValue(inputValue);
            }
            function handleSubmit(event) {
              event.preventDefault();
              // logic...
            }

            return React.createElement("form", {
              onSubmit: handleSubmit
            }, React.createElement(_form.BeyondInput, {
              type: "text",
              name: "name-error",
              label: "Insert a text...",
              onChange: handleChange,
              value: value,
              hasError: true,
              errorMessage: "Error in input"
            }));
          }
        }
      });

      /********************************************
      INTERNAL MODULE: ./views/result-input-loading
      ********************************************/

      ims.set('./views/result-input-loading', {
        hash: 1423872825,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultInputLoading = ResultInputLoading;
          var React = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultInputLoading() {
            const [value, setValue] = React.useState("");
            function handleChange(event) {
              const {
                value: inputValue
              } = event.target;
              setValue(inputValue);
            }
            ;
            function handleSubmit(event) {
              event.preventDefault();
              // logic...
            }

            ;
            return React.createElement("form", {
              onSubmit: handleSubmit
            }, React.createElement(_form.BeyondInput, {
              type: "text",
              name: "name-loading",
              label: "Insert a text...",
              onChange: handleChange,
              value: value,
              loading: true,
              colorSpinner: "#fff"
            }));
          }
        }
      });

      /*********************************************
      INTERNAL MODULE: ./views/result-input-password
      *********************************************/

      ims.set('./views/result-input-password', {
        hash: 414249406,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultInputPassword = ResultInputPassword;
          var React = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultInputPassword() {
            const [value, setValue] = React.useState("");
            function handleChange(event) {
              const {
                value: inputValue
              } = event.target;
              setValue(inputValue);
            }
            ;
            function handleSubmit(event) {
              event.preventDefault();
              // logic...
            }

            ;
            return React.createElement("form", {
              onSubmit: handleSubmit
            }, React.createElement(_form.BeyondInput, {
              type: "password",
              name: "name-password",
              label: "Insert a text...",
              onChange: handleChange,
              value: value,
              password: true
            }));
          }
        }
      });

      /**********************************************
      INTERNAL MODULE: ./views/result-input-with-icon
      **********************************************/

      ims.set('./views/result-input-with-icon', {
        hash: 3896870333,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.ResultInputWithIcon = ResultInputWithIcon;
          var React = require("react");
          var _form = require("@bgroup/ui/form");
          function ResultInputWithIcon() {
            const [value, setValue] = React.useState("");
            function handleChange(event) {
              const {
                value: inputValue
              } = event.target;
              setValue(inputValue);
            }
            ;
            function handleSubmit(event) {
              event.preventDefault();
              // logic...
            }

            ;
            return React.createElement("form", {
              onSubmit: handleSubmit
            }, React.createElement(_form.BeyondInput, {
              type: "text",
              name: "name-icon",
              label: "Insert a text...",
              onChange: handleChange,
              value: value,
              icon: "user"
            }));
          }
        }
      });

      /*******************************
      INTERNAL MODULE: ./views/results
      *******************************/

      ims.set('./views/results', {
        hash: 2756230774,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.Results = Results;
          var React = require("react");
          var _form = require("@bgroup/ui/form");
          function Results() {
            const [value, setValue] = React.useState('');
            function handleChange(event) {
              const {
                value: inputValue
              } = event.target;
              setValue(inputValue);
            }
            function handleSubmit(event) {
              event.preventDefault();
              // logic...
            }

            return React.createElement("form", {
              onSubmit: handleSubmit
            }, React.createElement(_form.BeyondInput, {
              type: "text",
              name: "name",
              label: "Insert a text...",
              onChange: handleChange,
              value: value
            }));
          }
        }
      });
      __pkg.exports.descriptor = [{
        "im": "./controller",
        "from": "Controller",
        "name": "Controller"
      }, {
        "im": "./views/index",
        "from": "View",
        "name": "View"
      }];
      // Module exports
      __pkg.exports.process = function ({
        require,
        prop,
        value
      }) {
        (require || prop === 'Controller') && _export("Controller", Controller = require ? require('./controller').Controller : value);
        (require || prop === 'View') && _export("View", View = require ? require('./views/index').View : value);
      };
      _export("__beyond_pkg", __beyond_pkg = __pkg);
      _export("hmr", hmr = new function () {
        this.on = (event, listener) => void 0;
        this.off = (event, listener) => void 0;
      }());
      __pkg.initialise(ims);
    }
  };
});
