// @bgroup/ui/beyond-input-controller/beyond-input-controller.amd.js
define(["exports", "module", "@beyond-js/widgets/render", "@beyond-js/kernel/bundle", "@beyond-js/react-widgets/base", "react", "@bgroup/ui/header", "@bgroup/ui/copyable-implementation", "@bgroup/ui/input-controller", "@beyond-js/kernel/styles"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5, dependency_6, dependency_7) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.Widget = _exports.Controller = void 0;
  const {
    Bundle: __Bundle
  } = dependency_1;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "@bgroup/ui@0.0.34/beyond-input-controller"
    },
    "type": "widget"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0], ['@beyond-js/react-widgets/base', dependency_2], ['react', dependency_3], ['@bgroup/ui/header', dependency_4], ['@bgroup/ui/copyable-implementation', dependency_5], ['@bgroup/ui/input-controller', dependency_6], ['@beyond-js/kernel/styles', dependency_7]]);
  brequire('@beyond-js/widgets/render').widgets.register([{
    "name": "beyond-input-controller",
    "vspecifier": "@bgroup/ui@0.0.34/beyond-input-controller",
    "is": "page",
    "route": "/components/beyond-input-controller"
  }]);
  brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-input-controller');
  const ims = new Map();

  /****************************
  INTERNAL MODULE: ./controller
  ****************************/

  ims.set('./controller', {
    hash: 8577895,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Controller = void 0;
      var _base = require("@beyond-js/react-widgets/base");
      var _views = require("./views");
      /*bundle*/
      class Controller extends _base.ReactWidgetController {
        get Widget() {
          return _views.Widget;
        }
      }
      exports.Controller = Controller;
    }
  });

  /**************************************
  INTERNAL MODULE: ./views/basic-template
  **************************************/

  ims.set('./views/basic-template', {
    hash: 1730820643,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.basicTemplate = void 0;
      const basicTemplate = [{
        label: 'Input radio',
        name: 'radio',
        events: {},
        options: [{
          type: 'radioItem',
          label: 'Male',
          value: '2'
        }, {
          type: 'radioItem',
          label: 'Woman',
          value: '1'
        }],
        placeholder: 'radio',
        className: 'input-control',
        data: null,
        type: 'radio'
      }, {
        label: 'Input checkbox',
        name: 'checkbox',
        events: {},
        options: [{
          type: 'checkItem',
          label: 'Argentina',
          value: '2'
        }, {
          type: 'checkItem',
          label: 'Brasil',
          value: '1'
        }],
        placeholder: 'checkbox',
        className: 'input-control',
        data: null,
        type: 'checkbox'
      }, {
        label: 'Input text',
        name: 'text',
        events: {},
        options: '',
        placeholder: 'text',
        className: 'input-control',
        data: null,
        type: 'text'
      }, {
        label: 'Input email',
        name: 'email',
        events: {},
        placeholder: 'email',
        className: 'input-control',
        data: null,
        type: 'email'
      }, {
        label: 'Input number',
        name: 'number',
        events: {},
        placeholder: 'number',
        className: 'input-control',
        data: null,
        type: 'number'
      }, {
        label: 'Input tel',
        name: 'tel',
        events: {},
        placeholder: 'tel',
        className: 'input-control',
        data: null,
        type: 'tel'
      }, {
        label: 'Input password',
        name: 'password',
        events: {},
        placeholder: 'password',
        className: 'input-control',
        data: null,
        type: 'password'
      }];
      exports.basicTemplate = basicTemplate;
    }
  });

  /***************************************
  INTERNAL MODULE: ./views/implementations
  ***************************************/

  ims.set('./views/implementations', {
    hash: 2462016764,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.template = exports.component = void 0;
      const component = `
<Component template={template} />
`;
      exports.component = component;
      const template = `
[
	{
		label: 'Input radio',
		name: 'radio',
		events: {},
		options: [
			{ type: 'radioItem', label: 'Male', value: '2' },
			{ type: 'radioItem', label: 'Woman', value: '1' },
		],
		placeholder: 'radio',
		className: 'input-control',
		data: null,
		type: 'radio',
	},

	{
		label: 'Input checkbox',
		name: 'checkbox',
		events: {},
		options: [
			{ type: 'checkItem', label: 'Argentina', value: '2' },
			{ type: 'checkItem', label: 'Brasil', value: '1' },
		],
		placeholder: 'checkbox',
		className: 'input-control',
		data: null,
		type: 'checkbox',
	},

	{
		label: 'Input text',
		name: 'text',
		events: {},
		options: '',
		placeholder: 'text',
		className: 'input-control',
		data: null,
		type: 'text',
	},
	{
		label: 'Input email',
		name: 'email',
		events: {},
		placeholder: 'email',
		className: 'input-control',
		data: null,
		type: 'email',
	},
	{
		label: 'Input number',
		name: 'number',
		events: {},
		placeholder: 'number',
		className: 'input-control',
		data: null,
		type: 'number',
	},
	{
		label: 'Input tel',
		name: 'tel',
		events: {},
		placeholder: 'tel',
		className: 'input-control',
		data: null,
		type: 'tel',
	},
	{
		label: 'Input password',
		name: 'password',
		events: {},
		placeholder: 'password',
		className: 'input-control',
		data: null,
		type: 'password',
	},
];`;
      exports.template = template;
    }
  });

  /*****************************
  INTERNAL MODULE: ./views/index
  *****************************/

  ims.set('./views/index', {
    hash: 2429979668,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Widget = Widget;
      var React = require("react");
      var _header = require("@bgroup/ui/header");
      var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
      var _implementations = require("./implementations");
      var _results = require("./results");
      /*bundle*/
      function Widget() {
        return React.createElement("div", {
          className: "view input-controller-view"
        }, React.createElement(_header.Header, {
          componentName: "Input Controller"
        }), React.createElement("div", {
          className: "component-info"
        }, React.createElement("h3", null, "Import: "), React.createElement(_copyableImplementation.CopyableImplementation, null, `import {InputController} from '@bgroup/ui/input-controller';`)), React.createElement("div", {
          className: "component-info"
        }, React.createElement("h3", null, "Implementation"), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementations.component)), React.createElement("div", {
          className: 'component-implementation'
        }, React.createElement("div", {
          className: "component-info"
        }, React.createElement("h3", null, "template.json"), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementations.template)), React.createElement("div", {
          className: "component-result"
        }, React.createElement("h3", null, "Result:"), React.createElement(_results.Results, null))));
      }
    }
  });

  /*******************************
  INTERNAL MODULE: ./views/results
  *******************************/

  ims.set('./views/results', {
    hash: 366821303,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Results = Results;
      var React = require("react");
      var _inputController = require("@bgroup/ui/input-controller");
      var _basicTemplate = require("./basic-template");
      function Results() {
        return React.createElement(_inputController.Component, {
          template: _basicTemplate.basicTemplate
        });
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./controller",
    "from": "Controller",
    "name": "Controller"
  }, {
    "im": "./views/index",
    "from": "Widget",
    "name": "Widget"
  }];
  let Controller, Widget;

  // Module exports
  _exports.Widget = Widget;
  _exports.Controller = Controller;
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'Controller') && (_exports.Controller = Controller = require ? require('./controller').Controller : value);
    (require || prop === 'Widget') && (_exports.Widget = Widget = require ? require('./views/index').Widget : value);
  };
  const __beyond_pkg = __pkg;
  _exports.__beyond_pkg = __beyond_pkg;
  const hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  _exports.hmr = hmr;
  __pkg.initialise(ims);
});
