// @bgroup/ui/home/home.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/kernel/styles';
import * as dependency_3 from '@beyond-js/react-widgets/page';
import * as dependency_4 from 'react';
import * as dependency_5 from '@bgroup/ui/header';
import * as dependency_6 from '@bgroup/ui/copyable-implementation';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/home"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/kernel/styles', dependency_2],['@beyond-js/react-widgets/page', dependency_3],['react', dependency_4],['@bgroup/ui/header', dependency_5],['@bgroup/ui/copyable-implementation', dependency_6]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"home-page","vspecifier":"@bgroup/ui@0.0.34/home","is":"page","route":"/"}]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/home')


const ims = new Map();

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 3327242745, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _page = require("@beyond-js/react-widgets/page");
var _views = require("./views");
/*bundle*/class Controller extends _page.PageReactWidgetController {
  get Widget() {
    return _views.Page;
  }
}
exports.Controller = Controller;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 483678332, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Page = Page;
var React = require("react");
var _header = require("@bgroup/ui/header");
var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
/*bundle*/function Page() {
  return React.createElement("div", {
    className: "view"
  }, React.createElement(_header.Header, {
    componentName: "Getting started"
  }), React.createElement("div", {
    className: "information"
  }, React.createElement("div", {
    className: "title"
  }, React.createElement("h3", null, "BEYOND ", React.createElement("span", {
    className: "ui"
  }, "UI"), " | Useful and reusable components")), React.createElement("p", null, "Beyond UI is a collection of reusable components here you can see what they are and how to import and implement them. "), React.createElement("p", null, "It provides a set of useful and highly customizable components to be used in your projects! "), React.createElement("div", {
    className: 'to-install'
  }, React.createElement("h4", null, "To Install"), React.createElement(_copyableImplementation.CopyableImplementation, null, `npm i @bgroup/ui`))));
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"},{"im":"./views/index","from":"Page","name":"Page"}];

export let Controller, Page;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);
    (require || prop === 'Page') && (Page = require ? require('./views/index').Page : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
