// @bgroup/ui/empty/empty.sjs.js
System.register(["@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react", "@bgroup/ui/icons"], function (_export, _context) {
  "use strict";

  var dependency_0, dependency_1, dependency_2, dependency_3, __Bundle, __pkg, ims, BeyondEmpty, __beyond_pkg, hmr;
  _export("BeyondEmpty", void 0);
  return {
    setters: [function (_beyondJsKernelBundle) {
      dependency_0 = _beyondJsKernelBundle;
    }, function (_beyondJsKernelStyles) {
      dependency_1 = _beyondJsKernelStyles;
    }, function (_react) {
      dependency_2 = _react;
    }, function (_bgroupUiIcons) {
      dependency_3 = _bgroupUiIcons;
    }],
    execute: function () {
      ({
        Bundle: __Bundle
      } = dependency_0);
      __pkg = new __Bundle({
        "module": {
          "vspecifier": "@bgroup/ui@0.0.34/empty"
        },
        "type": "code",
        "name": "empty"
      }, _context.meta.url).package();
      ;
      __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2], ['@bgroup/ui/icons', dependency_3]]);
      brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/empty');
      ims = new Map();
      /***********************
      INTERNAL MODULE: ./empty
      ***********************/
      ims.set('./empty', {
        hash: 1117662106,
        creator: function (require, exports) {
          "use strict";

          Object.defineProperty(exports, "__esModule", {
            value: true
          });
          exports.BeyondEmpty = BeyondEmpty;
          var React = require("react");
          var _icons = require("@bgroup/ui/icons");
          /*bundle*/
          function BeyondEmpty(props) {
            const text = props.text ? props.text : 'No hay registros';
            const cls = `${props.className ?? ''} beyond-element-empty`;
            return React.createElement("div", {
              className: cls
            }, React.createElement("div", {
              className: "content"
            }, props.icon && React.createElement(_icons.BeyondIcon, {
              icon: props.icon
            }), React.createElement("h3", null, text), props.additionalElement ? props.additionalElement : null, props.children ? props.children : null));
          }
        }
      });
      __pkg.exports.descriptor = [{
        "im": "./empty",
        "from": "BeyondEmpty",
        "name": "BeyondEmpty"
      }];
      // Module exports
      __pkg.exports.process = function ({
        require,
        prop,
        value
      }) {
        (require || prop === 'BeyondEmpty') && _export("BeyondEmpty", BeyondEmpty = require ? require('./empty').BeyondEmpty : value);
      };
      _export("__beyond_pkg", __beyond_pkg = __pkg);
      _export("hmr", hmr = new function () {
        this.on = (event, listener) => void 0;
        this.off = (event, listener) => void 0;
      }());
      __pkg.initialise(ims);
    }
  };
});
