// @bgroup/ui/empty/empty.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react", "@bgroup/ui/icons"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2, dependency_3) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.BeyondEmpty = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "@bgroup/ui@0.0.34/empty"
    },
    "type": "code",
    "name": "empty"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2], ['@bgroup/ui/icons', dependency_3]]);
  brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/empty');
  const ims = new Map();

  /***********************
  INTERNAL MODULE: ./empty
  ***********************/

  ims.set('./empty', {
    hash: 1117662106,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.BeyondEmpty = BeyondEmpty;
      var React = require("react");
      var _icons = require("@bgroup/ui/icons");
      /*bundle*/
      function BeyondEmpty(props) {
        const text = props.text ? props.text : 'No hay registros';
        const cls = `${props.className ?? ''} beyond-element-empty`;
        return React.createElement("div", {
          className: cls
        }, React.createElement("div", {
          className: "content"
        }, props.icon && React.createElement(_icons.BeyondIcon, {
          icon: props.icon
        }), React.createElement("h3", null, text), props.additionalElement ? props.additionalElement : null, props.children ? props.children : null));
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./empty",
    "from": "BeyondEmpty",
    "name": "BeyondEmpty"
  }];
  let BeyondEmpty;

  // Module exports
  _exports.BeyondEmpty = BeyondEmpty;
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'BeyondEmpty') && (_exports.BeyondEmpty = BeyondEmpty = require ? require('./empty').BeyondEmpty : value);
  };
  const __beyond_pkg = __pkg;
  _exports.__beyond_pkg = __beyond_pkg;
  const hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  _exports.hmr = hmr;
  __pkg.initialise(ims);
});
