// @bgroup/ui/beyond-swiper/beyond-swiper.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/react-widgets/page';
import * as dependency_3 from 'react';
import * as dependency_4 from '@bgroup/ui/header';
import * as dependency_5 from '@bgroup/ui/copyable-implementation';
import * as dependency_6 from '@bgroup/ui/swiper-component';
import * as dependency_7 from '@bgroup/ui/image';
import * as dependency_8 from '@beyond-js/kernel/styles';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/beyond-swiper"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/react-widgets/page', dependency_2],['react', dependency_3],['@bgroup/ui/header', dependency_4],['@bgroup/ui/copyable-implementation', dependency_5],['@bgroup/ui/swiper-component', dependency_6],['@bgroup/ui/image', dependency_7],['@beyond-js/kernel/styles', dependency_8]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"beyond-swiper","vspecifier":"@bgroup/ui@0.0.34/beyond-swiper","is":"page","route":"/components/beyond-swiper"}]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-swiper')


const ims = new Map();

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 3638196390, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _page = require("@beyond-js/react-widgets/page");
var _views = require("./views");
/*bundle*/class Controller extends _page.PageReactWidgetController {
  get Widget() {
    return _views.View;
  }
}
exports.Controller = Controller;
}});

/**************************************
INTERNAL MODULE: ./views/implementation
**************************************/

ims.set('./views/implementation', {hash: 1256036374, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.implementation = void 0;
const implementation = `
export function SwiperImplementation(): JSX.Element {
	const items = Array.from(Array(10)).map((_, index) => (
	  <BeyondImage src="https://source.unsplash.com/random" alt=" " />
	));
	return (
	  <BeyondSwiperSlider
		loop={true}
		slidesPerView="auto"
		spaceBetween={0}
		pagination={true}
		navigation={true}
		grabCursor={true}
	  >
		{items}
	  </BeyondSwiperSlider>
	);
  }
`;
exports.implementation = implementation;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 1328744491, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.View = View;
var React = require("react");
var _header = require("@bgroup/ui/header");
var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
var _result = require("./result");
var _implementation = require("./implementation");
/*bundle*/function View() {
  return React.createElement("div", {
    className: "view"
  }, React.createElement(_header.Header, {
    componentName: "Swiper"
  }), React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Import:"), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { BeyondSwiperSlider } from "@bgroup/ui/swiper-component"`)), React.createElement("div", {
    className: 'component-implementation'
  }, React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Implementation:"), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation)), React.createElement("div", {
    className: "component-result"
  }, React.createElement("h3", null, "Result:"), React.createElement(_result.Result, null))));
}
}});

/******************************
INTERNAL MODULE: ./views/result
******************************/

ims.set('./views/result', {hash: 1582697715, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Result = Result;
var _react = require("react");
var _swiperComponent = require("@bgroup/ui/swiper-component");
var _image = require("@bgroup/ui/image");
function Result() {
  const items = Array.from(Array(10)).map((_, index) => {
    return _react.default.createElement(_image.BeyondImage, {
      key: index.toString(),
      src: `https://source.unsplash.com/random/?city,night`,
      alt: " "
    });
  });
  return _react.default.createElement(_swiperComponent.BeyondSwiperSlider
  //  loop={true}
  , {
    //  loop={true}
    slidesPerView: "auto",
    spaceBetween: 20,
    pagination: true,
    navigation: true,
    className: "swiper-implementation"
  }, items);
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"},{"im":"./views/index","from":"View","name":"View"}];

export let Controller, View;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);
    (require || prop === 'View') && (View = require ? require('./views/index').View : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
