// @bgroup/ui/layout/main/layout/main.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/kernel/styles';
import * as dependency_3 from 'react';
import * as dependency_4 from '@bgroup/ui/icons';
import * as dependency_5 from '@beyond-js/kernel/routing';
import * as dependency_6 from '@bgroup/ui/perfect-scrollbar';
import * as dependency_7 from '@beyond-js/react-widgets/base';
import * as dependency_8 from '@bgroup/ui/toast';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/layout/main"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/kernel/styles', dependency_2],['react', dependency_3],['@bgroup/ui/icons', dependency_4],['@beyond-js/kernel/routing', dependency_5],['@bgroup/ui/perfect-scrollbar', dependency_6],['@beyond-js/react-widgets/base', dependency_7],['@bgroup/ui/toast', dependency_8]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"main-layout","vspecifier":"@bgroup/ui@0.0.34/layout/main","is":"layout"}]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/layout/main')


const ims = new Map();

/*******************************
INTERNAL MODULE: ./bg-components
*******************************/

ims.set('./bg-components', {hash: 2684328785, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.components = void 0;
const components = [{
  name: 'Beyond Empty',
  path: 'beyond-empty',
  subComponents: []
}, {
  name: 'Beyond Form',
  path: 'beyond-form',
  subComponents: [{
    name: 'Checkbox',
    path: 'beyond-checkbox'
  }, {
    name: 'Input',
    path: 'beyond-input'
  }, {
    name: 'Switch',
    path: 'beyond-switch'
  }, {
    name: 'Textarea',
    path: 'beyond-textarea'
  }, {
    name: 'Button',
    path: 'beyond-button'
  }, {
    name: 'Radio',
    path: 'beyond-radio'
  }, {
    name: 'Form',
    path: 'beyond-form'
  }]
}, {
  name: 'Beyond Modals',
  path: 'beyond-modals',
  subComponents: [{
    name: 'Alert',
    path: 'beyond-alert-modal'
  }, {
    name: 'Confirm',
    path: 'beyond-confirm-modal'
  }, {
    name: 'Regular',
    path: 'beyond-regular-modal'
  }]
}, {
  name: 'Beyond Icon',
  path: 'beyond-icons',
  subComponents: []
}, {
  name: 'Beyond Spinner',
  path: 'beyond-spinner',
  subComponents: []
}, {
  name: 'Beyond Input Controller',
  path: 'beyond-input-controller',
  subComponents: []
}, {
  name: 'Beyond Image',
  path: 'beyond-image',
  subComponents: []
}, {
  name: 'Beyond Scroll',
  path: 'beyond-scroll-container',
  subComponents: []
}, {
  name: 'Beyond Tabs',
  path: 'beyond-tabs',
  subComponents: []
}, {
  name: 'Beyond Toast',
  path: 'beyond-toast',
  subComponents: []
}, {
  name: 'Beyond Swiper',
  path: 'beyond-swiper',
  subComponents: []
}, {
  name: 'Beyond Verification Code',
  path: 'beyond-code-verification',
  subComponents: []
}, {
  name: 'Beyond Alert',
  path: 'beyond-alert',
  subComponents: []
}, {
  name: 'Drag And Drop',
  path: 'beyond-drag-and-drop',
  subComponents: []
}, {
  name: 'Link',
  path: 'beyond-link',
  subComponents: []
}];
exports.components = components;
}});

/*********************************************
INTERNAL MODULE: ./components/sidebar-dropdown
*********************************************/

ims.set('./components/sidebar-dropdown', {hash: 524109578, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SidebarDropdown = SidebarDropdown;
var React = require("react");
var _sidebarItem = require("./sidebar-item");
var _icons = require("@bgroup/ui/icons");
function SidebarDropdown({
  component
}) {
  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
  const subComponentsElements = component.subComponents.map(subComponent => React.createElement(_sidebarItem.SidebarItem, {
    key: subComponent.path,
    component: subComponent
  }));
  const iconDirection = isDropdownOpen ? 'down' : 'right';
  function toggleDropdown(event) {
    event.preventDefault();
    setIsDropdownOpen(!isDropdownOpen);
  }
  return React.createElement("li", {
    className: "sidebar-item"
  }, React.createElement("a", {
    onClick: toggleDropdown
  }, React.createElement(_icons.BeyondIcon, {
    icon: "arrowDropDown",
    className: iconDirection
  }), component.name), isDropdownOpen && React.createElement("ul", {
    className: "sub-list"
  }, subComponentsElements));
}
}});

/*****************************************
INTERNAL MODULE: ./components/sidebar-item
*****************************************/

ims.set('./components/sidebar-item', {hash: 1856212488, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SidebarItem = SidebarItem;
var React = require("react");
var _routing = require("@beyond-js/kernel/routing");
function SidebarItem({
  component
}) {
  const location = `/components/${component.path}`;
  function redirect(event) {
    event.preventDefault();
    _routing.routing.pushState(location);
  }
  return React.createElement("li", {
    className: "sidebar-item"
  }, React.createElement("a", {
    onClick: redirect
  }, component.name));
}
}});

/************************************
INTERNAL MODULE: ./components/sidebar
************************************/

ims.set('./components/sidebar', {hash: 1979810423, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sidebar = Sidebar;
var React = require("react");
var _bgComponents = require("../bg-components");
var _sidebarDropdown = require("./sidebar-dropdown");
var _sidebarItem = require("./sidebar-item");
var _routing = require("@beyond-js/kernel/routing");
var _perfectScrollbar = require("@bgroup/ui/perfect-scrollbar");
function Sidebar() {
  const componentsElements = _bgComponents.components.map(component => {
    if (component?.subComponents.length > 0) {
      return React.createElement(_sidebarDropdown.SidebarDropdown, {
        key: component.path,
        component: component
      });
    }
    return React.createElement(_sidebarItem.SidebarItem, {
      key: component.path,
      component: component
    });
  });
  function goHome(event) {
    event.preventDefault();
    _routing.routing.pushState("/");
  }
  return React.createElement(_perfectScrollbar.BeyondScrollContainer, {
    className: "aside",
    "data-perfect-scrollbar": "",
    "data-suppress-scroll-x": "true"
  }, React.createElement("aside", null, React.createElement("h1", {
    className: "logo"
  }, React.createElement("a", {
    onClick: goHome,
    href: "/"
  }, "Beyond ", React.createElement("span", null, "UI"))), React.createElement("div", {
    className: "nav-wrap"
  }, React.createElement("nav", {
    className: "main-nav",
    role: "navigation"
  }, React.createElement("ul", {
    className: "unstyled list-hover-slide"
  }, componentsElements)))), React.createElement("div", {
    className: "ps__rail-x"
  }, React.createElement("div", {
    className: "ps__thumb-x",
    tabIndex: 0
  })), React.createElement("div", {
    className: "ps__rail-y"
  }, React.createElement("div", {
    className: " ps__thumb-y",
    tabIndex: 0
  })));
}
}});

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 2684072206, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _base = require("@beyond-js/react-widgets/base");
var _views = require("./views");
/*bundle*/class Controller extends _base.ReactWidgetController {
  get Widget() {
    return _views.Layout;
  }
}
exports.Controller = Controller;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 989967076, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Layout = Layout;
var React = require("react");
var _sidebar = require("../components/sidebar");
var _toast = require("@bgroup/ui/toast");
function Layout() {
  return React.createElement("main", null, React.createElement(_sidebar.Sidebar, null), React.createElement(_toast.Toasts, {
    position: {
      bottom: '2rem',
      right: '2rem'
    }
  }), React.createElement("div", {
    className: "main-content"
  }, React.createElement("beyond-layout-children", null)));
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"}];

export let Controller;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
