// @bgroup/ui/reactive-model/reactive-model.browser.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from '@beyond-js/kernel/core';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/reactive-model"},"type":"ts"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/kernel/core', dependency_1]]);

const ims = new Map();

/********************************
INTERNAL MODULE: ./reactive-model
********************************/

ims.set('./reactive-model', {hash: 3765919911, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ReactiveModel = void 0;
var _core = require("@beyond-js/kernel/core");
/*bundle*/class ReactiveModel extends _core.Events {
  _ready;
  get ready() {
    return this._ready;
  }
  _fetching;
  get fetching() {
    return this._fetching;
  }
  set fetching(value) {
    if (value === this._fetching) return;
    this._fetching = value;
    this.triggerEvent();
  }
  _fetched;
  get fetched() {
    return this._fetched;
  }
  _processing;
  get processing() {
    return this._processing;
  }
  _processed;
  get processed() {
    return this._processed;
  }
  _loaded;
  get loaded() {
    return this._loaded;
  }
  triggerEvent = (event = 'change') => this.trigger(event);
  /**
   * set value in a private property
   * @param property
   * @param value
   */
  _set(property, value) {
    let props = {};
    if (property && value !== 'undefined') props[property] = value;else props = property;
    let updated = false;
    /* A comment. */
    for (const prop in props) {
      const key = `_${prop}`;
      if (!this.hasOwnProperty(key)) continue; //same value on store
      if (this[key] === props[prop]) continue;
      this[key] = props[prop];
      updated = true;
    }
    if (updated) this.triggerEvent();
  }
  getProperties() {
    const props = {};
    Object.keys(this).forEach(property => props[property.replace('_', '')] = this[property]);
    return props;
  }
}
exports.ReactiveModel = ReactiveModel;
}});

__pkg.exports.descriptor = [{"im":"./reactive-model","from":"ReactiveModel","name":"ReactiveModel"}];

export let ReactiveModel;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'ReactiveModel') && (ReactiveModel = require ? require('./reactive-model').ReactiveModel : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
