// @bgroup/ui/toast/toast.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react", "@bgroup/ui/hooks", "@bgroup/ui/reactive-model", "@bgroup/ui/icons"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.toast = _exports.hmr = _exports.__beyond_pkg = _exports.Toasts = _exports.ToastTypes = _exports.Toast = _exports.IToast = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "@bgroup/ui@0.0.34/toast"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2], ['@bgroup/ui/hooks', dependency_3], ['@bgroup/ui/reactive-model', dependency_4], ['@bgroup/ui/icons', dependency_5]]);
  brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/toast');
  const ims = new Map();

  /***********************
  INTERNAL MODULE: ./index
  ***********************/

  ims.set('./index', {
    hash: 3366825821,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Toasts = Toasts;
      var _react = require("react");
      var _hooks = require("@bgroup/ui/hooks");
      var _model = require("./model");
      var _toast = require("./toast");
      /*bundle*/
      function Toasts({
        position = {
          bottom: "1rem",
          right: "1rem"
        },
        className,
        ...props
      }) {
        const [items, setItems] = _react.default.useState([]);
        (0, _hooks.useBinder)([_model.toast], () => setItems(_model.toast.current), "current-toasts-changed");
        const elements = items.map(item => _react.default.createElement(_toast.Toast, {
          key: item.id,
          ...item
        }));
        return _react.default.createElement("div", {
          style: position,
          className: `essential__toasts ${className}`,
          ...props
        }, elements);
      }
    }
  });

  /***********************
  INTERNAL MODULE: ./model
  ***********************/

  ims.set('./model', {
    hash: 3981715842,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.toast = void 0;
      var _reactiveModel = require("@bgroup/ui/reactive-model");
      class Toast extends _reactiveModel.ReactiveModel {
        #current;
        get current() {
          return this.#current;
        }
        set current(newValue) {
          this.#current = newValue;
          this.triggerEvent('current-toasts-changed');
        }
        constructor() {
          super();
          this.#current = [];
        }
        #generateUniqueId() {
          return crypto.getRandomValues(new Uint8Array(16)).toString();
        }
        #add(type, message, duration) {
          const newToast = {
            id: this.#generateUniqueId(),
            message,
            type,
            duration
          };
          this.#current = [...this.#current, newToast];
          this.triggerEvent('current-toasts-changed');
          return newToast.id;
        }
        remove(toastId) {
          this.#current = this.#current.filter(toast => toast.id !== toastId);
          this.triggerEvent('current-toasts-changed');
        }
        success(message, duration) {
          return this.#add('success', message, duration);
        }
        error(message, duration) {
          return this.#add('error', message, duration);
        }
        warning(message, duration) {
          return this.#add('warning', message, duration);
        }
        info(message, duration) {
          return this.#add('info', message, duration);
        }
        loading(message, duration) {
          return this.#add('loading', message, duration);
        }
      }
      /*bundle*/
      const toast = new Toast();
      exports.toast = toast;
    }
  });

  /***********************
  INTERNAL MODULE: ./toast
  ***********************/

  ims.set('./toast', {
    hash: 1589754565,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Toast = Toast;
      var _react = require("react");
      var _model = require("./model");
      var _icons = require("@bgroup/ui/icons");
      const DEFAULT_DURATION = 3000;
      const ANIMATION_MARGIN = 300;
      /*bundle*/
      function Toast({
        type,
        message,
        duration,
        id
      }) {
        const toastRef = _react.default.useRef(null);
        _react.default.useEffect(() => {
          if (toastRef.current) {
            toastRef.current.classList.add('enter');
          }
          return () => {
            if (toastRef.current) toastRef.current.classList.remove('enter');
          };
        }, [toastRef.current]);
        _react.default.useEffect(() => {
          duration = duration ?? DEFAULT_DURATION;
          setTimeout(() => {
            toastRef.current.classList.remove('enter');
            toastRef.current.classList.add('exit');
          }, duration);
          const timeout = setTimeout(() => {
            _model.toast.current = _model.toast.current.filter(item => item.id !== id);
          }, duration + ANIMATION_MARGIN);
          return () => clearTimeout(timeout);
        }, [duration, id]);
        const icons = {
          error: "closeCircle",
          success: "tickCircle",
          info: "infoCircle",
          loading: "refreshCircle",
          warning: "infoCircle"
        };
        return _react.default.createElement("article", {
          ref: toastRef,
          className: `toast ${type}`
        }, _react.default.createElement(_icons.BeyondIcon, {
          icon: icons[type],
          className: "icon"
        }), _react.default.createElement("p", {
          className: "message"
        }, message));
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./index",
    "from": "Toasts",
    "name": "Toasts"
  }, {
    "im": "./model",
    "from": "ToastTypes",
    "name": "ToastTypes"
  }, {
    "im": "./model",
    "from": "IToast",
    "name": "IToast"
  }, {
    "im": "./model",
    "from": "toast",
    "name": "toast"
  }, {
    "im": "./toast",
    "from": "Toast",
    "name": "Toast"
  }];
  let Toasts, ToastTypes, IToast, toast, Toast;

  // Module exports
  _exports.Toast = Toast;
  _exports.toast = toast;
  _exports.IToast = IToast;
  _exports.ToastTypes = ToastTypes;
  _exports.Toasts = Toasts;
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'Toasts') && (_exports.Toasts = Toasts = require ? require('./index').Toasts : value);
    (require || prop === 'ToastTypes') && (_exports.ToastTypes = ToastTypes = require ? require('./model').ToastTypes : value);
    (require || prop === 'IToast') && (_exports.IToast = IToast = require ? require('./model').IToast : value);
    (require || prop === 'toast') && (_exports.toast = toast = require ? require('./model').toast : value);
    (require || prop === 'Toast') && (_exports.Toast = Toast = require ? require('./toast').Toast : value);
  };
  const __beyond_pkg = __pkg;
  _exports.__beyond_pkg = __beyond_pkg;
  const hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  _exports.hmr = hmr;
  __pkg.initialise(ims);
});
