// @bgroup/ui/empty/empty.browser.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from '@beyond-js/kernel/styles';
import * as dependency_2 from 'react';
import * as dependency_3 from '@bgroup/ui/icons';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/empty"},"type":"code","name":"empty"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1],['react', dependency_2],['@bgroup/ui/icons', dependency_3]]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/empty')


const ims = new Map();

/***********************
INTERNAL MODULE: ./empty
***********************/

ims.set('./empty', {hash: 1117662106, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BeyondEmpty = BeyondEmpty;
var React = require("react");
var _icons = require("@bgroup/ui/icons");
/*bundle*/function BeyondEmpty(props) {
  const text = props.text ? props.text : 'No hay registros';
  const cls = `${props.className ?? ''} beyond-element-empty`;
  return React.createElement("div", {
    className: cls
  }, React.createElement("div", {
    className: "content"
  }, props.icon && React.createElement(_icons.BeyondIcon, {
    icon: props.icon
  }), React.createElement("h3", null, text), props.additionalElement ? props.additionalElement : null, props.children ? props.children : null));
}
}});

__pkg.exports.descriptor = [{"im":"./empty","from":"BeyondEmpty","name":"BeyondEmpty"}];

export let BeyondEmpty;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'BeyondEmpty') && (BeyondEmpty = require ? require('./empty').BeyondEmpty : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
