// @bgroup/ui/beyond-link/beyond-link.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/react-widgets/page';
import * as dependency_3 from 'react';
import * as dependency_4 from '@bgroup/ui/header';
import * as dependency_5 from '@bgroup/ui/copyable-implementation';
import * as dependency_6 from '@bgroup/ui/link';
import * as dependency_7 from '@beyond-js/kernel/styles';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/beyond-link"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/react-widgets/page', dependency_2],['react', dependency_3],['@bgroup/ui/header', dependency_4],['@bgroup/ui/copyable-implementation', dependency_5],['@bgroup/ui/link', dependency_6],['@beyond-js/kernel/styles', dependency_7]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"beyond-link-code","vspecifier":"@bgroup/ui@0.0.34/beyond-link","is":"page","route":"/components/beyond-link"}]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-link')


const ims = new Map();

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 3638196390, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _page = require("@beyond-js/react-widgets/page");
var _views = require("./views");
/*bundle*/class Controller extends _page.PageReactWidgetController {
  get Widget() {
    return _views.View;
  }
}
exports.Controller = Controller;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 3344396871, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.View = View;
var React = require("react");
var _header = require("@bgroup/ui/header");
var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
var _result = require("./result");
/*bundle*/function View() {
  return React.createElement("div", {
    className: "view"
  }, React.createElement(_header.Header, {
    componentName: "Link"
  }), React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Import:"), React.createElement(_copyableImplementation.CopyableImplementation, null, `import { Link } from '@bgroup/ui/link';`)), React.createElement("div", {
    className: 'component-implementation'
  }, React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Implementation:"), React.createElement(_copyableImplementation.CopyableImplementation, null, `<Link href='/' >This is a link to home</Link>`), React.createElement("div", {
    className: 'component-detail'
  }, React.createElement("h4", null, "Additional Information:"), React.createElement("div", null, React.createElement("p", null, "If you want to execute a function before the redirection you can pass an ", React.createElement("code", null, "onClick"), " which will be executed before the redirection.")))), React.createElement("div", {
    className: "component-result"
  }, React.createElement("h3", null, "Result:"), React.createElement(_result.Result, null))));
}
}});

/******************************
INTERNAL MODULE: ./views/result
******************************/

ims.set('./views/result', {hash: 3109461048, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Result = Result;
var _react = require("react");
var _link = require("@bgroup/ui/link");
function Result() {
  return _react.default.createElement(_link.Link, {
    href: '/'
  }, "This is a link to home");
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"},{"im":"./views/index","from":"View","name":"View"}];

export let Controller, View;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);
    (require || prop === 'View') && (View = require ? require('./views/index').View : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
