// @bgroup/ui/beyond-radio/beyond-radio.browser.mjs
import * as dependency_0 from '@beyond-js/widgets/render';
import * as dependency_1 from '@beyond-js/kernel/bundle';
import * as dependency_2 from '@beyond-js/react-widgets/page';
import * as dependency_3 from 'react';
import * as dependency_4 from '@bgroup/ui/header';
import * as dependency_5 from '@bgroup/ui/copyable-implementation';
import * as dependency_6 from '@bgroup/ui/form';
import * as dependency_7 from '@beyond-js/kernel/styles';

const {Bundle: __Bundle} = dependency_1;
const __pkg = new __Bundle({"module":{"vspecifier":"@bgroup/ui@0.0.34/beyond-radio"},"type":"widget"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/widgets/render', dependency_0],['@beyond-js/react-widgets/page', dependency_2],['react', dependency_3],['@bgroup/ui/header', dependency_4],['@bgroup/ui/copyable-implementation', dependency_5],['@bgroup/ui/form', dependency_6],['@beyond-js/kernel/styles', dependency_7]]);

brequire('@beyond-js/widgets/render').widgets.register([{"name":"beyond-radio","vspecifier":"@bgroup/ui@0.0.34/beyond-radio","is":"page","route":"/components/beyond-radio"}]);

brequire('@beyond-js/kernel/styles').styles.register('@bgroup/ui@0.0.34/beyond-radio')


const ims = new Map();

/****************************
INTERNAL MODULE: ./controller
****************************/

ims.set('./controller', {hash: 3638196390, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Controller = void 0;
var _page = require("@beyond-js/react-widgets/page");
var _views = require("./views");
/*bundle*/class Controller extends _page.PageReactWidgetController {
  get Widget() {
    return _views.View;
  }
}
exports.Controller = Controller;
}});

/**************************************
INTERNAL MODULE: ./views/implementation
**************************************/

ims.set('./views/implementation', {hash: 219735980, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.implementation = void 0;
const implementation = `
export function CheckboxImplementation() {
	const [selectedValue, setSelectedValue] = React.useState('radio-1');
	
	function handleChange(event) {
		const { value } = event.target;
		setSelectedValue(value);
	};

	return (
		<form className="results-form">
			<BeyondRadio
				onChange={handleChange}
				checked={selectedValue === 'radio-1'}
				name='name'
				value="radio-1"
				label="Radio 1"
			/>
			<BeyondRadio
				onChange={handleChange}
				checked={selectedValue === 'radio-2'}
				name='name'
				value="radio-2"
				label="Radio 2"
			/>
		</form>
	);
}`;
exports.implementation = implementation;
}});

/*****************************
INTERNAL MODULE: ./views/index
*****************************/

ims.set('./views/index', {hash: 3723373121, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.View = View;
var React = require("react");
var _header = require("@bgroup/ui/header");
var _copyableImplementation = require("@bgroup/ui/copyable-implementation");
var _results = require("./results");
var _implementation = require("./implementation");
/*bundle*/function View() {
  return React.createElement("div", {
    className: "view"
  }, React.createElement(_header.Header, {
    componentName: "Radio"
  }), React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Import: "), React.createElement(_copyableImplementation.CopyableImplementation, null, `import {BeyondRadio} from '@bgroup/ui/form';`)), React.createElement("div", {
    className: 'component-implementation'
  }, React.createElement("div", {
    className: "component-info"
  }, React.createElement("h3", null, "Implementation:"), React.createElement(_copyableImplementation.CopyableImplementation, null, _implementation.implementation)), React.createElement("div", {
    className: "component-result"
  }, React.createElement("h3", null, "Result:"), React.createElement(_results.Results, null))));
}
}});

/*******************************
INTERNAL MODULE: ./views/results
*******************************/

ims.set('./views/results', {hash: 3371236537, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Results = Results;
var React = require("react");
var _form = require("@bgroup/ui/form");
function Results() {
  const [selectedValue, setSelectedValue] = React.useState('radio-1');
  function handleChange(event) {
    const {
      value
    } = event.target;
    setSelectedValue(value);
  }
  ;
  return React.createElement("form", {
    className: "results-form"
  }, React.createElement(_form.BeyondRadio, {
    onChange: handleChange,
    checked: selectedValue === 'radio-1',
    name: 'name',
    value: "radio-1",
    label: "Radio 1"
  }), React.createElement(_form.BeyondRadio, {
    onChange: handleChange,
    checked: selectedValue === 'radio-2',
    name: 'name',
    value: "radio-2",
    label: "Radio 2"
  }));
}
}});

__pkg.exports.descriptor = [{"im":"./controller","from":"Controller","name":"Controller"},{"im":"./views/index","from":"View","name":"View"}];

export let Controller, View;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Controller') && (Controller = require ? require('./controller').Controller : value);
    (require || prop === 'View') && (View = require ? require('./views/index').View : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
